{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "eb05d447-e0a0-4f30-9931-9c7e2fad8ef0",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: RNOmni\n",
      "\n",
      "Registered S3 methods overwritten by 'treeio':\n",
      "  method              from    \n",
      "  MRCA.phylo          tidytree\n",
      "  MRCA.treedata       tidytree\n",
      "  Nnode.treedata      tidytree\n",
      "  Ntip.treedata       tidytree\n",
      "  ancestor.phylo      tidytree\n",
      "  ancestor.treedata   tidytree\n",
      "  child.phylo         tidytree\n",
      "  child.treedata      tidytree\n",
      "  full_join.phylo     tidytree\n",
      "  full_join.treedata  tidytree\n",
      "  groupClade.phylo    tidytree\n",
      "  groupClade.treedata tidytree\n",
      "  groupOTU.phylo      tidytree\n",
      "  groupOTU.treedata   tidytree\n",
      "  inner_join.phylo    tidytree\n",
      "  inner_join.treedata tidytree\n",
      "  is.rooted.treedata  tidytree\n",
      "  nodeid.phylo        tidytree\n",
      "  nodeid.treedata     tidytree\n",
      "  nodelab.phylo       tidytree\n",
      "  nodelab.treedata    tidytree\n",
      "  offspring.phylo     tidytree\n",
      "  offspring.treedata  tidytree\n",
      "  parent.phylo        tidytree\n",
      "  parent.treedata     tidytree\n",
      "  root.treedata       tidytree\n",
      "  rootnode.phylo      tidytree\n",
      "  sibling.phylo       tidytree\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(EWCE) # Versions prior to 1.5.5 did not compute controlled results properly\n",
    "library(orthogene)\n",
    "library(Seurat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e531558-f996-433a-b3c2-df02dd9fb088",
   "metadata": {},
   "source": [
    "### setup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c333e9a1-d0cc-4d3e-97dc-88761f0d4ca7",
   "metadata": {
    "scrolled": true,
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "96"
      ],
      "text/latex": [
       "96"
      ],
      "text/markdown": [
       "96"
      ],
      "text/plain": [
       "[1] 96"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "wd <- '~/codebases/MacBrainDev/'\n",
    "data.dir <- 'data/'\n",
    "setwd(wd)\n",
    "\n",
    "\n",
    "n.cores <-  as.numeric(Sys.getenv('SLURM_CPUS_PER_TASK'))\n",
    "n.cores <- if (!is.na(n.cores) & n.cores > 1) n.cores else parallel::detectCores()\n",
    "n.cores\n",
    "\n",
    "options(future.globals.maxSize= 5*1024^3)\n",
    "future::plan(strategy = \"multicore\", workers = n.cores)\n",
    "\n",
    "disease_lists <- readRDS(glue::glue('{data.dir}Disease_genes/all_diseases_list.rds'))\n",
    "\n",
    "# For markers only\n",
    "downsample_ncells_per_subtype <- Inf\n",
    "\n",
    "# Define filenames\n",
    "\n",
    "base.name <- 'All.MNN.v1.org.fct'\n",
    "\n",
    "indata.fname <- paste0(data.dir, base.name, '.rds')\n",
    "# ctd object\n",
    "ctd.fname <- paste0(data.dir, 'ewce_ctd.', base.name,\n",
    "                    '.ds', downsample_ncells_per_subtype, '.rds')\n",
    "# markers\n",
    "markers.fname <- paste0(data.dir, 'ewce_downsampled_markers.',\n",
    "                        base.name,  '.ds', \n",
    "                        downsample_ncells_per_subtype,'.rds')\n",
    "# sct object\n",
    "sct.fname <- paste0(data.dir, 'ewce_sct.', base.name, '.rds')\n",
    "# sct normalized object\n",
    "sct.normed.fname <- paste0(data.dir, 'ewce_sct_normed_markers_orthos.', base.name, \n",
    "                           '.ds', downsample_ncells_per_subtype, '.rds')\n",
    "\n",
    "# Define parameters\n",
    "# for ewce\n",
    "reps <- 20000 #  >10000\n",
    "verbose <- T\n",
    "\n",
    "# ewce\n",
    "dis.res.folder <- paste0(data.dir, 'EWCE_diseases.ds', downsample_ncells_per_subtype, '/')\n",
    "dir.create(dis.res.folder, showWarnings = F)\n",
    "EWCE_results.fname <- paste0('EWCE_results.', reps, '_reps.ds', downsample_ncells_per_subtype, '.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d0aa2743-2ef5-4a8e-9d72-e8efbeb1b4ae",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: SeuratObject\n",
      "\n",
      "Loading required package: sp\n",
      "\n",
      "The legacy packages maptools, rgdal, and rgeos, underpinning the sp package,\n",
      "which was just loaded, will retire in October 2023.\n",
      "Please refer to R-spatial evolution reports for details, especially\n",
      "https://r-spatial.org/r/2023/05/15/evolution4.html.\n",
      "It may be desirable to make the sf package available;\n",
      "package maintainers should consider adding sf to Suggests:.\n",
      "The sp package is now running under evolution status 2\n",
      "     (status 2 uses the sf package in place of rgdal)\n",
      "\n",
      "Loading required package: Seurat\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "An object of class Seurat \n",
       "34619 features across 761529 samples within 1 assay \n",
       "Active assay: RNA (34619 features, 0 variable features)\n",
       " 2 dimensional reductions calculated: mnn, umap"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Load data\n",
    "indata <- readRDS(indata.fname)\n",
    "# Print summary\n",
    "indata\n",
    "\n",
    "Idents(indata) <- 'subtype'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d180c01e-7cd0-4704-b54e-c426b55a7c64",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Find All Marker genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "772b50f5-8a2e-48f1-82af-f5dae14d43c0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/users/genomics/xoel/codebases/MacBrainDev/data/ewce_downsampled_markers.All.MNN.v1.org.fct.dsInf.rds'"
      ],
      "text/latex": [
       "'/users/genomics/xoel/codebases/MacBrainDev/data/ewce\\_downsampled\\_markers.All.MNN.v1.org.fct.dsInf.rds'"
      ],
      "text/markdown": [
       "'/users/genomics/xoel/codebases/MacBrainDev/data/ewce_downsampled_markers.All.MNN.v1.org.fct.dsInf.rds'"
      ],
      "text/plain": [
       "[1] \"/users/genomics/xoel/codebases/MacBrainDev/data/ewce_downsampled_markers.All.MNN.v1.org.fct.dsInf.rds\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "normalizePath(markers.fname)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "57c5d8aa-88fd-4f04-9888-f03779acfeb6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'data/ewce_downsampled_markers.All.MNN.v1.org.fct.dsInf.rds'"
      ],
      "text/latex": [
       "'data/ewce\\_downsampled\\_markers.All.MNN.v1.org.fct.dsInf.rds'"
      ],
      "text/markdown": [
       "'data/ewce_downsampled_markers.All.MNN.v1.org.fct.dsInf.rds'"
      ],
      "text/plain": [
       "[1] \"data/ewce_downsampled_markers.All.MNN.v1.org.fct.dsInf.rds\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "9146"
      ],
      "text/latex": [
       "9146"
      ],
      "text/markdown": [
       "9146"
      ],
      "text/plain": [
       "[1] 9146"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2503"
      ],
      "text/latex": [
       "2503"
      ],
      "text/markdown": [
       "2503"
      ],
      "text/plain": [
       "[1] 2503"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "519"
      ],
      "text/latex": [
       "519"
      ],
      "text/markdown": [
       "519"
      ],
      "text/plain": [
       "[1] 519"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "markers.fname\n",
    "\n",
    "# Check if run\n",
    "if (!file.exists(markers.fname) & !file.exists(ctd.fname)){\n",
    "    \n",
    "    print('Computing Markers')\n",
    "    \n",
    "    Idents(indata) <- 'subtype'\n",
    "    \n",
    "    all.markers <- FindAllMarkers(\n",
    "      # ds.seurat,\n",
    "      indata,\n",
    "      assay = 'RNA',\n",
    "      features = NULL,\n",
    "      logfc.threshold = 0.25,\n",
    "      test.use = \"wilcox\",\n",
    "      slot = \"data\",\n",
    "      min.pct = 0.1,\n",
    "      min.diff.pct = -Inf,\n",
    "      node = NULL,\n",
    "      verbose = TRUE,\n",
    "      only.pos = FALSE,\n",
    "      max.cells.per.ident = downsample_ncells_per_subtype,\n",
    "      random.seed = 1,\n",
    "      latent.vars = NULL,\n",
    "      min.cells.feature = 3,\n",
    "      min.cells.group = 3,\n",
    "      pseudocount.use = 1,\n",
    "      mean.fxn = NULL,\n",
    "      fc.name = NULL,\n",
    "      base = 2,\n",
    "      return.thresh = 0.01,\n",
    "      densify = FALSE,\n",
    "    )\n",
    "    \n",
    "    saveRDS(all.markers, markers.fname)\n",
    "    \n",
    "} else {\n",
    "    all.markers <- readRDS(markers.fname)\n",
    "}\n",
    "\n",
    "# ADJUSTED P-VALUE < 0.05\n",
    "length(unique(dplyr::filter(all.markers, p_val_adj < 0.05)$gene))\n",
    "\n",
    "# ADJUSTED P-VALUE < 0.05 & |AVG log2 FC| > 1 \n",
    "length(unique(dplyr::filter(all.markers, p_val_adj < 0.05, abs(avg_log2FC) > 1)$gene)) \n",
    "\n",
    "# ADJUSTED P-VALUE < 0.05 & |AVG log2 FC| > 2\n",
    "length(unique(dplyr::filter(all.markers, p_val_adj < 0.05, abs(avg_log2FC) > 2)$gene)) \n",
    "\n",
    "significant.markers <- unique(dplyr::filter(all.markers, p_val_adj < 0.05)$gene)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dfde3693-09e3-4ccc-8971-ef6bba5d2f91",
   "metadata": {},
   "source": [
    "### SCT Normalization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c7c9fbd4-72f6-49e2-9c68-e40c07e20568",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'data/ewce_ctd.All.MNN.v1.org.fct.dsInf.rds'"
      ],
      "text/latex": [
       "'data/ewce\\_ctd.All.MNN.v1.org.fct.dsInf.rds'"
      ],
      "text/markdown": [
       "'data/ewce_ctd.All.MNN.v1.org.fct.dsInf.rds'"
      ],
      "text/plain": [
       "[1] \"data/ewce_ctd.All.MNN.v1.org.fct.dsInf.rds\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Lower number of cores to reduce memory usagefuture::plan(strategy = \"multicore\", workers = min(40, n.cores))\n",
    "\n",
    "file.exists(sct.normed.fname)\n",
    "ctd.fname\n",
    "file.exists(ctd.fname)\n",
    "# If CTD is already computed, we don't need anything here\n",
    "if (file.exists(sct.normed.fname) & !file.exists(ctd.fname)){\n",
    "    exp.normed <- readRDS(sct.normed.fname)\n",
    "    \n",
    "} else if (!file.exists(ctd.fname)){\n",
    "    \n",
    "    # If we already have the sct-data, we can load it, otherwise we run it\n",
    "    if (file.exists(sct.fname)){\n",
    "        sct <- readRDS(sct.fname)\n",
    "    } else {\n",
    "        requireNamespace(\"sctransform\")\n",
    "\n",
    "        raw.expression <- indata[['RNA']]@counts\n",
    "        # First, perform sctransform of data\n",
    "        sct <- sctransform::vst(umi = raw.expression,\n",
    "                                method=\"glmGamPoi\", \n",
    "                                return_cell_attr = TRUE, \n",
    "                                n_genes = 3000,\n",
    "                                verbosity = 2)\n",
    "        saveRDS(sct, sct.fname)\n",
    "    }\n",
    "    \n",
    "    ## Get orthologues data\n",
    "    # Check number of genes in SCT-data and in significant markers\n",
    "    print(paste('Number of genes before SCT:', nrow(indata)))\n",
    "    sct.genes <- rownames(sct$y)\n",
    "    print(paste('Number of genes after SCT:', length(sct.genes)))\n",
    "    print(paste('Number of marker genes:', length(significant.markers)))\n",
    "    # Intersect lists\n",
    "    markers.sct.genes <- sct.genes[sct.genes %in% significant.markers]\n",
    "    print(paste('Number of genes after SCT in markers:', length(markers.sct.genes)))\n",
    "    # Convert macaque to human genes\n",
    "    orthos <- convert_orthologs(markers.sct.genes,input_species = 'macaque', verbose = F)\n",
    "    print(paste('Number of matching orthologs:', nrow(orthos)))\n",
    "\n",
    "    ## Correct orthologues data\n",
    "    # Get subset of data\n",
    "    ortho.raw <- indata[['RNA']]@counts[orthos$input_gene,]\n",
    "\n",
    "    # Correct subset data using full model\n",
    "    ortho.sct <- sctransform::correct_counts(x = sct, umi = ortho.raw, verbosity = 2)\n",
    "    # Rename genes in object\n",
    "    rownames(ortho.sct) <- rownames(orthos)\n",
    "    # Normalize gene expression\n",
    "    exp.normed <- Matrix::t(Matrix::t(ortho.sct) * (1/Matrix::colSums(ortho.sct)))\n",
    "    \n",
    "    saveRDS(exp.normed, sct.normed.fname)\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1b12d1f4-1d0d-42ea-b89d-eb1f4a314766",
   "metadata": {},
   "source": [
    "### CellTypeData object"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "aa651139-633f-4fce-9668-a02f650901b6",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# increase number of cores for parallel computing\n",
    "future::plan(strategy = \"multicore\", workers = min(80, n.cores))\n",
    "\n",
    "# Define annotation levels\n",
    "annot.levels <- as.list(indata@meta.data[,c('subclass','subtype')])\n",
    "\n",
    "if (file.exists(ctd.fname)){\n",
    "\n",
    "    ctd <- ctd.fname\n",
    "\n",
    "} else {\n",
    "\n",
    "    ctd <- generate_celltype_data(\n",
    "    \n",
    "    exp=exp.normed,\n",
    "    annotLevels=annot.levels,\n",
    "    groupName='MacaqueDev',\n",
    "    \n",
    "    input_species='hsapiens',\n",
    "    output_species='hsapiens',\n",
    "    convert_orths=F,\n",
    "    \n",
    "    specificity_quantiles=T,\n",
    "    dendrograms=T,\n",
    "    \n",
    "    file_prefix=paste0('ewce_ctd.',base.name),\n",
    "    force_new_file=T,\n",
    "    \n",
    "    no_cores=n.cores,\n",
    "    as_sparse = TRUE,\n",
    "    as_DelayedArray = FALSE,\n",
    "    verbose=T)\n",
    "    \n",
    "    file.copy(ctd, ctd.fname)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef6b1b83-e388-431d-9527-ad853db35340",
   "metadata": {},
   "source": [
    "### Perform analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "764460a8-5962-4fad-a1ed-cff82cdb6aab",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"2023-10-13 15:46:21 CEST\"\n"
     ]
    }
   ],
   "source": [
    "print(Sys.time())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "f8adb040-596e-4192-90c0-f13078a362cc",
   "metadata": {
    "scrolled": true,
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>ASD HC65</dt><dd>65</dd><dt>DD</dt><dd>93</dd><dt>SFARI Score1</dt><dd>207</dd><dt>SFARI Score2</dt><dd>211</dd><dt>SFARI Score3</dt><dd>506</dd><dt>SFARI Syndromic</dt><dd>126</dd><dt>GWAS::ADHD 2019</dt><dd>38</dd><dt>GWAS::AD 2019</dt><dd>192</dd><dt>GWAS::AN 2019</dt><dd>107</dd><dt>GWAS::ASD 2019</dt><dd>29</dd><dt>GWAS::BD 2019</dt><dd>218</dd><dt>GWAS::IQ 2018</dt><dd>859</dd><dt>GWAS::MDD 2018</dt><dd>69</dd><dt>GWAS::NEUROT 2018</dt><dd>300</dd><dt>GWAS::PD 2014</dt><dd>34</dd><dt>GWAS::SCZ 2020</dt><dd>130</dd><dt>MAGMA::ADHD 2019</dt><dd>200</dd><dt>MAGMA::AD 2019</dt><dd>200</dd><dt>MAGMA::AN 2019</dt><dd>200</dd><dt>MAGMA::ASD 2019</dt><dd>200</dd><dt>MAGMA::BD 2019</dt><dd>200</dd><dt>MAGMA::IQ 2018</dt><dd>200</dd><dt>MAGMA::MDD 2018</dt><dd>200</dd><dt>MAGMA::NEUROT 2018</dt><dd>200</dd><dt>MAGMA::OCD 2018</dt><dd>200</dd><dt>MAGMA::PD 2014</dt><dd>200</dd><dt>MAGMA::SCZ 2020</dt><dd>200</dd><dt>MAGMA::TS 2019</dt><dd>200</dd><dt>DISGENET::Giant Cell Glioblastoma</dt><dd>84</dd><dt>DISGENET::Glioblastoma</dt><dd>79</dd><dt>DISGENET::Glioblastoma Multiforme</dt><dd>111</dd><dt>DISGENET::Glioma</dt><dd>87</dd><dt>DISGENET::M.Oligoastr+Oligodendrogliomas</dt><dd>32</dd><dt>DISGENET::Malignant Glioma</dt><dd>70</dd><dt>DISGENET::Medulloblastomas</dt><dd>50</dd><dt>DISGENET::Neuroblastoma</dt><dd>47</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[ASD HC65] 65\n",
       "\\item[DD] 93\n",
       "\\item[SFARI Score1] 207\n",
       "\\item[SFARI Score2] 211\n",
       "\\item[SFARI Score3] 506\n",
       "\\item[SFARI Syndromic] 126\n",
       "\\item[GWAS::ADHD 2019] 38\n",
       "\\item[GWAS::AD 2019] 192\n",
       "\\item[GWAS::AN 2019] 107\n",
       "\\item[GWAS::ASD 2019] 29\n",
       "\\item[GWAS::BD 2019] 218\n",
       "\\item[GWAS::IQ 2018] 859\n",
       "\\item[GWAS::MDD 2018] 69\n",
       "\\item[GWAS::NEUROT 2018] 300\n",
       "\\item[GWAS::PD 2014] 34\n",
       "\\item[GWAS::SCZ 2020] 130\n",
       "\\item[MAGMA::ADHD 2019] 200\n",
       "\\item[MAGMA::AD 2019] 200\n",
       "\\item[MAGMA::AN 2019] 200\n",
       "\\item[MAGMA::ASD 2019] 200\n",
       "\\item[MAGMA::BD 2019] 200\n",
       "\\item[MAGMA::IQ 2018] 200\n",
       "\\item[MAGMA::MDD 2018] 200\n",
       "\\item[MAGMA::NEUROT 2018] 200\n",
       "\\item[MAGMA::OCD 2018] 200\n",
       "\\item[MAGMA::PD 2014] 200\n",
       "\\item[MAGMA::SCZ 2020] 200\n",
       "\\item[MAGMA::TS 2019] 200\n",
       "\\item[DISGENET::Giant Cell Glioblastoma] 84\n",
       "\\item[DISGENET::Glioblastoma] 79\n",
       "\\item[DISGENET::Glioblastoma Multiforme] 111\n",
       "\\item[DISGENET::Glioma] 87\n",
       "\\item[DISGENET::M.Oligoastr+Oligodendrogliomas] 32\n",
       "\\item[DISGENET::Malignant Glioma] 70\n",
       "\\item[DISGENET::Medulloblastomas] 50\n",
       "\\item[DISGENET::Neuroblastoma] 47\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "ASD HC65\n",
       ":   65DD\n",
       ":   93SFARI Score1\n",
       ":   207SFARI Score2\n",
       ":   211SFARI Score3\n",
       ":   506SFARI Syndromic\n",
       ":   126GWAS::ADHD 2019\n",
       ":   38GWAS::AD 2019\n",
       ":   192GWAS::AN 2019\n",
       ":   107GWAS::ASD 2019\n",
       ":   29GWAS::BD 2019\n",
       ":   218GWAS::IQ 2018\n",
       ":   859GWAS::MDD 2018\n",
       ":   69GWAS::NEUROT 2018\n",
       ":   300GWAS::PD 2014\n",
       ":   34GWAS::SCZ 2020\n",
       ":   130MAGMA::ADHD 2019\n",
       ":   200MAGMA::AD 2019\n",
       ":   200MAGMA::AN 2019\n",
       ":   200MAGMA::ASD 2019\n",
       ":   200MAGMA::BD 2019\n",
       ":   200MAGMA::IQ 2018\n",
       ":   200MAGMA::MDD 2018\n",
       ":   200MAGMA::NEUROT 2018\n",
       ":   200MAGMA::OCD 2018\n",
       ":   200MAGMA::PD 2014\n",
       ":   200MAGMA::SCZ 2020\n",
       ":   200MAGMA::TS 2019\n",
       ":   200DISGENET::Giant Cell Glioblastoma\n",
       ":   84DISGENET::Glioblastoma\n",
       ":   79DISGENET::Glioblastoma Multiforme\n",
       ":   111DISGENET::Glioma\n",
       ":   87DISGENET::M.Oligoastr+Oligodendrogliomas\n",
       ":   32DISGENET::Malignant Glioma\n",
       ":   70DISGENET::Medulloblastomas\n",
       ":   50DISGENET::Neuroblastoma\n",
       ":   47\n",
       "\n"
      ],
      "text/plain": [
       "                                ASD HC65 \n",
       "                                      65 \n",
       "                                      DD \n",
       "                                      93 \n",
       "                            SFARI Score1 \n",
       "                                     207 \n",
       "                            SFARI Score2 \n",
       "                                     211 \n",
       "                            SFARI Score3 \n",
       "                                     506 \n",
       "                         SFARI Syndromic \n",
       "                                     126 \n",
       "                         GWAS::ADHD 2019 \n",
       "                                      38 \n",
       "                           GWAS::AD 2019 \n",
       "                                     192 \n",
       "                           GWAS::AN 2019 \n",
       "                                     107 \n",
       "                          GWAS::ASD 2019 \n",
       "                                      29 \n",
       "                           GWAS::BD 2019 \n",
       "                                     218 \n",
       "                           GWAS::IQ 2018 \n",
       "                                     859 \n",
       "                          GWAS::MDD 2018 \n",
       "                                      69 \n",
       "                       GWAS::NEUROT 2018 \n",
       "                                     300 \n",
       "                           GWAS::PD 2014 \n",
       "                                      34 \n",
       "                          GWAS::SCZ 2020 \n",
       "                                     130 \n",
       "                        MAGMA::ADHD 2019 \n",
       "                                     200 \n",
       "                          MAGMA::AD 2019 \n",
       "                                     200 \n",
       "                          MAGMA::AN 2019 \n",
       "                                     200 \n",
       "                         MAGMA::ASD 2019 \n",
       "                                     200 \n",
       "                          MAGMA::BD 2019 \n",
       "                                     200 \n",
       "                          MAGMA::IQ 2018 \n",
       "                                     200 \n",
       "                         MAGMA::MDD 2018 \n",
       "                                     200 \n",
       "                      MAGMA::NEUROT 2018 \n",
       "                                     200 \n",
       "                         MAGMA::OCD 2018 \n",
       "                                     200 \n",
       "                          MAGMA::PD 2014 \n",
       "                                     200 \n",
       "                         MAGMA::SCZ 2020 \n",
       "                                     200 \n",
       "                          MAGMA::TS 2019 \n",
       "                                     200 \n",
       "       DISGENET::Giant Cell Glioblastoma \n",
       "                                      84 \n",
       "                  DISGENET::Glioblastoma \n",
       "                                      79 \n",
       "       DISGENET::Glioblastoma Multiforme \n",
       "                                     111 \n",
       "                        DISGENET::Glioma \n",
       "                                      87 \n",
       "DISGENET::M.Oligoastr+Oligodendrogliomas \n",
       "                                      32 \n",
       "              DISGENET::Malignant Glioma \n",
       "                                      70 \n",
       "              DISGENET::Medulloblastomas \n",
       "                                      50 \n",
       "                 DISGENET::Neuroblastoma \n",
       "                                      47 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"1/36\"\n",
      "[1] \"### Disease: ASD HC65\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"1/36\"\n",
      "[1] \"### Disease: ASD HC65\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"2/36\"\n",
      "[1] \"### Disease: DD\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"2/36\"\n",
      "[1] \"### Disease: DD\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"3/36\"\n",
      "[1] \"### Disease: SFARI Score1\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"3/36\"\n",
      "[1] \"### Disease: SFARI Score1\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"4/36\"\n",
      "[1] \"### Disease: SFARI Score2\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"4/36\"\n",
      "[1] \"### Disease: SFARI Score2\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"5/36\"\n",
      "[1] \"### Disease: SFARI Score3\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"5/36\"\n",
      "[1] \"### Disease: SFARI Score3\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"6/36\"\n",
      "[1] \"### Disease: SFARI Syndromic\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"6/36\"\n",
      "[1] \"### Disease: SFARI Syndromic\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"7/36\"\n",
      "[1] \"### Disease: GWAS::ADHD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"7/36\"\n",
      "[1] \"### Disease: GWAS::ADHD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"8/36\"\n",
      "[1] \"### Disease: GWAS::AD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"8/36\"\n",
      "[1] \"### Disease: GWAS::AD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"9/36\"\n",
      "[1] \"### Disease: GWAS::AN 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"9/36\"\n",
      "[1] \"### Disease: GWAS::AN 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"10/36\"\n",
      "[1] \"### Disease: GWAS::ASD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"10/36\"\n",
      "[1] \"### Disease: GWAS::ASD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"11/36\"\n",
      "[1] \"### Disease: GWAS::BD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"11/36\"\n",
      "[1] \"### Disease: GWAS::BD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"12/36\"\n",
      "[1] \"### Disease: GWAS::IQ 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"12/36\"\n",
      "[1] \"### Disease: GWAS::IQ 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"13/36\"\n",
      "[1] \"### Disease: GWAS::MDD 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"13/36\"\n",
      "[1] \"### Disease: GWAS::MDD 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"14/36\"\n",
      "[1] \"### Disease: GWAS::NEUROT 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"14/36\"\n",
      "[1] \"### Disease: GWAS::NEUROT 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"15/36\"\n",
      "[1] \"### Disease: GWAS::PD 2014\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"15/36\"\n",
      "[1] \"### Disease: GWAS::PD 2014\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"16/36\"\n",
      "[1] \"### Disease: GWAS::SCZ 2020\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"16/36\"\n",
      "[1] \"### Disease: GWAS::SCZ 2020\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"17/36\"\n",
      "[1] \"### Disease: MAGMA::ADHD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"17/36\"\n",
      "[1] \"### Disease: MAGMA::ADHD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"18/36\"\n",
      "[1] \"### Disease: MAGMA::AD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"18/36\"\n",
      "[1] \"### Disease: MAGMA::AD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"19/36\"\n",
      "[1] \"### Disease: MAGMA::AN 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"19/36\"\n",
      "[1] \"### Disease: MAGMA::AN 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"20/36\"\n",
      "[1] \"### Disease: MAGMA::ASD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"20/36\"\n",
      "[1] \"### Disease: MAGMA::ASD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"21/36\"\n",
      "[1] \"### Disease: MAGMA::BD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"21/36\"\n",
      "[1] \"### Disease: MAGMA::BD 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"22/36\"\n",
      "[1] \"### Disease: MAGMA::IQ 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"22/36\"\n",
      "[1] \"### Disease: MAGMA::IQ 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"23/36\"\n",
      "[1] \"### Disease: MAGMA::MDD 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"23/36\"\n",
      "[1] \"### Disease: MAGMA::MDD 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"24/36\"\n",
      "[1] \"### Disease: MAGMA::NEUROT 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"24/36\"\n",
      "[1] \"### Disease: MAGMA::NEUROT 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"25/36\"\n",
      "[1] \"### Disease: MAGMA::OCD 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"25/36\"\n",
      "[1] \"### Disease: MAGMA::OCD 2018\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"26/36\"\n",
      "[1] \"### Disease: MAGMA::PD 2014\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"26/36\"\n",
      "[1] \"### Disease: MAGMA::PD 2014\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"27/36\"\n",
      "[1] \"### Disease: MAGMA::SCZ 2020\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"27/36\"\n",
      "[1] \"### Disease: MAGMA::SCZ 2020\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"28/36\"\n",
      "[1] \"### Disease: MAGMA::TS 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"28/36\"\n",
      "[1] \"### Disease: MAGMA::TS 2019\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"29/36\"\n",
      "[1] \"### Disease: DISGENET::Giant Cell Glioblastoma\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"29/36\"\n",
      "[1] \"### Disease: DISGENET::Giant Cell Glioblastoma\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"30/36\"\n",
      "[1] \"### Disease: DISGENET::Glioblastoma\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"30/36\"\n",
      "[1] \"### Disease: DISGENET::Glioblastoma\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"31/36\"\n",
      "[1] \"### Disease: DISGENET::Glioblastoma Multiforme\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"31/36\"\n",
      "[1] \"### Disease: DISGENET::Glioblastoma Multiforme\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"32/36\"\n",
      "[1] \"### Disease: DISGENET::Glioma\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"32/36\"\n",
      "[1] \"### Disease: DISGENET::Glioma\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"33/36\"\n",
      "[1] \"### Disease: DISGENET::M.Oligoastr+Oligodendrogliomas\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"33/36\"\n",
      "[1] \"### Disease: DISGENET::M.Oligoastr+Oligodendrogliomas\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"34/36\"\n",
      "[1] \"### Disease: DISGENET::Malignant Glioma\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"34/36\"\n",
      "[1] \"### Disease: DISGENET::Malignant Glioma\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"35/36\"\n",
      "[1] \"### Disease: DISGENET::Medulloblastomas\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"35/36\"\n",
      "[1] \"### Disease: DISGENET::Medulloblastomas\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"36/36\"\n",
      "[1] \"### Disease: DISGENET::Neuroblastoma\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 1\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n",
      "[1] \"36/36\"\n",
      "[1] \"### Disease: DISGENET::Neuroblastoma\"\n",
      "[1] \"### GC&Length control: FALSE\"\n",
      "[1] \"### Level: 2\"\n",
      "[1] \"Loading results\"\n",
      "[1] \"next...\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4572 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>CellType</th><th scope=col>annotLevel</th><th scope=col>p</th><th scope=col>fold_change</th><th scope=col>sd_from_mean</th><th scope=col>q</th><th scope=col>hit.cells</th><th scope=col>geneSizeControl</th><th scope=col>list</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Excitatory_neurons  </td><td>1</td><td>0.00005</td><td>1.6210245</td><td> 4.949783</td><td>0.00075000</td><td>5.1582406</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Inhibitory_neurons  </td><td>1</td><td>0.00025</td><td>1.6011505</td><td> 4.651712</td><td>0.00187500</td><td>4.7708344</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>gIPC                </td><td>1</td><td>0.05595</td><td>1.1455764</td><td> 1.676427</td><td>0.27975000</td><td>3.7065873</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>4</th><td><span style=white-space:pre-wrap>OPC&amp;Oligo           </span></td><td>1</td><td>0.09410</td><td>1.1895992</td><td> 1.392667</td><td>0.28470000</td><td>4.5192011</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>CR                  </td><td>1</td><td>0.09490</td><td>1.1910740</td><td> 1.366746</td><td>0.28470000</td><td>3.8939663</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>enIPC               </td><td>1</td><td>0.11695</td><td>1.1109784</td><td> 1.144413</td><td>0.29237500</td><td>2.7927076</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>Astro               </td><td>1</td><td>0.30115</td><td>1.0635441</td><td> 0.460351</td><td>0.64532143</td><td>3.9141276</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>inIPC               </td><td>1</td><td>0.87095</td><td>0.8844217</td><td>-1.075973</td><td>0.99990000</td><td>2.4041449</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>9</th><td><span style=white-space:pre-wrap>RB&amp;Vas              </span></td><td>1</td><td>0.91495</td><td>0.7682445</td><td>-1.260677</td><td>0.99990000</td><td>2.9470457</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>Immune              </td><td>1</td><td>0.97145</td><td>0.6909600</td><td>-1.608298</td><td>0.99990000</td><td>2.2556185</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>11</th><td>Mes                 </td><td>1</td><td>0.97440</td><td>0.7195762</td><td>-1.577106</td><td>0.99990000</td><td>2.1893353</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>12</th><td>PAT_related_subtypes</td><td>1</td><td>0.99620</td><td>0.7863180</td><td>-2.327396</td><td>0.99990000</td><td>2.2326491</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>GE_NSC              </td><td>1</td><td>0.99890</td><td>0.7601149</td><td>-2.664843</td><td>0.99990000</td><td>2.3555541</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>14</th><td>dorsal_NSC          </td><td>1</td><td>0.99980</td><td>0.7300410</td><td>-3.089397</td><td>0.99990000</td><td>2.0313232</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>15</th><td>Patterning_centers  </td><td>1</td><td>0.99990</td><td>0.6629693</td><td>-3.137240</td><td>0.99990000</td><td>1.8286641</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>16</th><td>InN_NR2F2_SP8       </td><td>2</td><td>0.00085</td><td>1.4028790</td><td> 3.810475</td><td>0.04666667</td><td>0.4731544</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>17</th><td>InN_NR2F2_CRH       </td><td>2</td><td>0.00130</td><td>1.5360277</td><td> 3.909490</td><td>0.04666667</td><td>0.6167353</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>18</th><td>InN_LHX6_MAF        </td><td>2</td><td>0.00135</td><td>1.4953935</td><td> 3.932042</td><td>0.04666667</td><td>0.4736134</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>19</th><td>ExN_CUX2_PALMD      </td><td>2</td><td>0.00170</td><td>1.4938997</td><td> 4.199320</td><td>0.04666667</td><td>0.5322803</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>20</th><td>InN_LHX6_SST_RELN   </td><td>2</td><td>0.00245</td><td>1.5609166</td><td> 3.993785</td><td>0.04666667</td><td>0.5792726</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>21</th><td>InN_LHX6_CCK        </td><td>2</td><td>0.00250</td><td>1.4522059</td><td> 3.749336</td><td>0.04666667</td><td>0.5620510</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>22</th><td>ExN_SOX5_OPRK1_SULF1</td><td>2</td><td>0.00300</td><td>1.9609908</td><td> 4.065703</td><td>0.04800000</td><td>1.1837654</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>23</th><td>ExN_SOX5_ID2        </td><td>2</td><td>0.00350</td><td>1.4128909</td><td> 3.286578</td><td>0.04900000</td><td>0.4937898</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>24</th><td>ExN_CUX2_ADRA2A     </td><td>2</td><td>0.00530</td><td>1.4537623</td><td> 3.388206</td><td>0.05740000</td><td>0.6081105</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>25</th><td>ExN_SOX5_PALMD      </td><td>2</td><td>0.00565</td><td>1.4095423</td><td> 3.271409</td><td>0.05740000</td><td>0.4244082</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>26</th><td>InN_SP8_PAX6        </td><td>2</td><td>0.00610</td><td>1.4349994</td><td> 3.283874</td><td>0.05740000</td><td>0.4753021</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>27</th><td>InN_SP8_CRH         </td><td>2</td><td>0.00615</td><td>1.5619382</td><td> 3.697464</td><td>0.05740000</td><td>0.6605383</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>28</th><td>ExN_SOX5_SYT6       </td><td>2</td><td>0.00695</td><td>1.4583315</td><td> 3.115525</td><td>0.05987692</td><td>0.5573811</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>29</th><td>ExN_CUX2_ACTN2      </td><td>2</td><td>0.00800</td><td>1.4812210</td><td> 3.141101</td><td>0.06160000</td><td>0.6831864</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>30</th><td>InN_LHX6_DCN        </td><td>2</td><td>0.00825</td><td>1.4817412</td><td> 3.385368</td><td>0.06160000</td><td>0.5407997</td><td>FALSE</td><td>ASD HC65</td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>4543</th><td>InN_NKX2_1_CCND2      </td><td>2</td><td>0.79680</td><td>0.8517079</td><td>-0.8025895</td><td>0.9651216</td><td>0.1490730</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4544</th><td>InN_SP8_MEIS2_RND3    </td><td>2</td><td>0.80190</td><td>0.8777507</td><td>-0.8198304</td><td>0.9651216</td><td>0.1538165</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4545</th><td>InN_LHX6_CRABP1       </td><td>2</td><td>0.80340</td><td>0.8372672</td><td>-0.5947048</td><td>0.9651216</td><td>0.2010952</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4546</th><td>oIPC_DLL1             </td><td>2</td><td>0.80530</td><td>0.8526240</td><td>-0.7419515</td><td>0.9651216</td><td>0.1958917</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4547</th><td>RB_HBA1_HBE1          </td><td>2</td><td>0.80855</td><td>0.5806570</td><td>-0.5761092</td><td>0.9651216</td><td>0.1073040</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4548</th><td>InN_SP8_PAX6          </td><td>2</td><td>0.82090</td><td>0.8485480</td><td>-0.8738893</td><td>0.9651216</td><td>0.1615550</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4549</th><td>InN_NR2F2_VIP         </td><td>2</td><td>0.82375</td><td>0.8023415</td><td>-0.8025472</td><td>0.9651216</td><td>0.2467083</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4550</th><td>Cls_FGF17_LGI1        </td><td>2</td><td>0.82770</td><td>0.8245840</td><td>-0.7245079</td><td>0.9651216</td><td>0.1882434</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4551</th><td>Astro_GFAP            </td><td>2</td><td>0.82920</td><td>0.7804154</td><td>-0.8778759</td><td>0.9651216</td><td>0.2374910</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4552</th><td>InN_NR2F2_LAMP5       </td><td>2</td><td>0.83095</td><td>0.8157384</td><td>-0.8706271</td><td>0.9651216</td><td>0.2165941</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4553</th><td>IPC_RSPO3_NEUROG1     </td><td>2</td><td>0.83300</td><td>0.7089259</td><td>-0.5987480</td><td>0.9651216</td><td>0.1830551</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4554</th><td>InN_FOXP2_TSHZ1       </td><td>2</td><td>0.85075</td><td>0.8382707</td><td>-0.9043727</td><td>0.9651216</td><td>0.1836489</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4555</th><td>Oligo_MBP_PLP1        </td><td>2</td><td>0.85480</td><td>0.6431800</td><td>-0.7476012</td><td>0.9651216</td><td>0.2308018</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4556</th><td>ExN_SOX5_KCNV1        </td><td>2</td><td>0.85610</td><td>0.8114204</td><td>-1.0291166</td><td>0.9651216</td><td>0.1999997</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4557</th><td>IPC_EOMES_VIM         </td><td>2</td><td>0.85810</td><td>0.8592431</td><td>-1.0250441</td><td>0.9651216</td><td>0.1505864</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4558</th><td>VLMC_PTGDS_DSP        </td><td>2</td><td>0.86025</td><td>0.6000721</td><td>-0.8290546</td><td>0.9651216</td><td>0.2154650</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4559</th><td>Astro_MFGE8           </td><td>2</td><td>0.86705</td><td>0.7452408</td><td>-0.9995434</td><td>0.9651216</td><td>0.2294424</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4560</th><td>IPC_EOMES_NHLH1_deep  </td><td>2</td><td>0.86935</td><td>0.8234710</td><td>-0.9620617</td><td>0.9651216</td><td>0.1365066</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4561</th><td>ExN_SOX5_SYT6         </td><td>2</td><td>0.87635</td><td>0.7978923</td><td>-1.0443020</td><td>0.9651216</td><td>0.1755591</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4562</th><td>ExN_CUX2_ADRA2A       </td><td>2</td><td>0.87895</td><td>0.8161150</td><td>-1.0427608</td><td>0.9651216</td><td>0.1961770</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4563</th><td>RB_HBA1_SNCA          </td><td>2</td><td>0.90390</td><td>0.7036430</td><td>-0.6623377</td><td>0.9828816</td><td>0.1466737</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4564</th><td>OPC_PDGFRA            </td><td>2</td><td>0.92540</td><td>0.7469174</td><td>-1.0876794</td><td>0.9834811</td><td>0.2139045</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4565</th><td>ExN_TSHZ2_NR4A3       </td><td>2</td><td>0.94015</td><td>0.6767839</td><td>-1.1731889</td><td>0.9834811</td><td>0.2261227</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4566</th><td>ExN_CUX2_PALMD        </td><td>2</td><td>0.94735</td><td>0.7858237</td><td>-1.3829181</td><td>0.9834811</td><td>0.1609537</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4567</th><td>PC_TTR                </td><td>2</td><td>0.94895</td><td>0.6603859</td><td>-0.8729626</td><td>0.9834811</td><td>0.1570861</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4568</th><td>ExN_SOX5_NR4A2_GRID2  </td><td>2</td><td>0.95240</td><td>0.7019206</td><td>-1.2645338</td><td>0.9834811</td><td>0.1675807</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4569</th><td>Endothelial_CLDN5_FLT1</td><td>2</td><td>0.96145</td><td>0.4904144</td><td>-1.1765550</td><td>0.9834811</td><td>0.1695690</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4570</th><td>Microglia_PTPRC_C1QC  </td><td>2</td><td>0.97050</td><td>0.4924781</td><td>-1.0450169</td><td>0.9834811</td><td>0.1539820</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4571</th><td>IPC_EOMES_NHLH1_up    </td><td>2</td><td>0.97470</td><td>0.7381648</td><td>-1.4672831</td><td>0.9834811</td><td>0.1388052</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "\t<tr><th scope=row>4572</th><td>Ependymal_FOXJ1       </td><td>2</td><td>0.98990</td><td>0.4851367</td><td>-1.0553920</td><td>0.9899000</td><td>0.1896320</td><td>FALSE</td><td>DISGENET::Neuroblastoma</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4572 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & CellType & annotLevel & p & fold\\_change & sd\\_from\\_mean & q & hit.cells & geneSizeControl & list\\\\\n",
       "  & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & Excitatory\\_neurons   & 1 & 0.00005 & 1.6210245 &  4.949783 & 0.00075000 & 5.1582406 & FALSE & ASD HC65\\\\\n",
       "\t2 & Inhibitory\\_neurons   & 1 & 0.00025 & 1.6011505 &  4.651712 & 0.00187500 & 4.7708344 & FALSE & ASD HC65\\\\\n",
       "\t3 & gIPC                 & 1 & 0.05595 & 1.1455764 &  1.676427 & 0.27975000 & 3.7065873 & FALSE & ASD HC65\\\\\n",
       "\t4 & OPC\\&Oligo            & 1 & 0.09410 & 1.1895992 &  1.392667 & 0.28470000 & 4.5192011 & FALSE & ASD HC65\\\\\n",
       "\t5 & CR                   & 1 & 0.09490 & 1.1910740 &  1.366746 & 0.28470000 & 3.8939663 & FALSE & ASD HC65\\\\\n",
       "\t6 & enIPC                & 1 & 0.11695 & 1.1109784 &  1.144413 & 0.29237500 & 2.7927076 & FALSE & ASD HC65\\\\\n",
       "\t7 & Astro                & 1 & 0.30115 & 1.0635441 &  0.460351 & 0.64532143 & 3.9141276 & FALSE & ASD HC65\\\\\n",
       "\t8 & inIPC                & 1 & 0.87095 & 0.8844217 & -1.075973 & 0.99990000 & 2.4041449 & FALSE & ASD HC65\\\\\n",
       "\t9 & RB\\&Vas               & 1 & 0.91495 & 0.7682445 & -1.260677 & 0.99990000 & 2.9470457 & FALSE & ASD HC65\\\\\n",
       "\t10 & Immune               & 1 & 0.97145 & 0.6909600 & -1.608298 & 0.99990000 & 2.2556185 & FALSE & ASD HC65\\\\\n",
       "\t11 & Mes                  & 1 & 0.97440 & 0.7195762 & -1.577106 & 0.99990000 & 2.1893353 & FALSE & ASD HC65\\\\\n",
       "\t12 & PAT\\_related\\_subtypes & 1 & 0.99620 & 0.7863180 & -2.327396 & 0.99990000 & 2.2326491 & FALSE & ASD HC65\\\\\n",
       "\t13 & GE\\_NSC               & 1 & 0.99890 & 0.7601149 & -2.664843 & 0.99990000 & 2.3555541 & FALSE & ASD HC65\\\\\n",
       "\t14 & dorsal\\_NSC           & 1 & 0.99980 & 0.7300410 & -3.089397 & 0.99990000 & 2.0313232 & FALSE & ASD HC65\\\\\n",
       "\t15 & Patterning\\_centers   & 1 & 0.99990 & 0.6629693 & -3.137240 & 0.99990000 & 1.8286641 & FALSE & ASD HC65\\\\\n",
       "\t16 & InN\\_NR2F2\\_SP8        & 2 & 0.00085 & 1.4028790 &  3.810475 & 0.04666667 & 0.4731544 & FALSE & ASD HC65\\\\\n",
       "\t17 & InN\\_NR2F2\\_CRH        & 2 & 0.00130 & 1.5360277 &  3.909490 & 0.04666667 & 0.6167353 & FALSE & ASD HC65\\\\\n",
       "\t18 & InN\\_LHX6\\_MAF         & 2 & 0.00135 & 1.4953935 &  3.932042 & 0.04666667 & 0.4736134 & FALSE & ASD HC65\\\\\n",
       "\t19 & ExN\\_CUX2\\_PALMD       & 2 & 0.00170 & 1.4938997 &  4.199320 & 0.04666667 & 0.5322803 & FALSE & ASD HC65\\\\\n",
       "\t20 & InN\\_LHX6\\_SST\\_RELN    & 2 & 0.00245 & 1.5609166 &  3.993785 & 0.04666667 & 0.5792726 & FALSE & ASD HC65\\\\\n",
       "\t21 & InN\\_LHX6\\_CCK         & 2 & 0.00250 & 1.4522059 &  3.749336 & 0.04666667 & 0.5620510 & FALSE & ASD HC65\\\\\n",
       "\t22 & ExN\\_SOX5\\_OPRK1\\_SULF1 & 2 & 0.00300 & 1.9609908 &  4.065703 & 0.04800000 & 1.1837654 & FALSE & ASD HC65\\\\\n",
       "\t23 & ExN\\_SOX5\\_ID2         & 2 & 0.00350 & 1.4128909 &  3.286578 & 0.04900000 & 0.4937898 & FALSE & ASD HC65\\\\\n",
       "\t24 & ExN\\_CUX2\\_ADRA2A      & 2 & 0.00530 & 1.4537623 &  3.388206 & 0.05740000 & 0.6081105 & FALSE & ASD HC65\\\\\n",
       "\t25 & ExN\\_SOX5\\_PALMD       & 2 & 0.00565 & 1.4095423 &  3.271409 & 0.05740000 & 0.4244082 & FALSE & ASD HC65\\\\\n",
       "\t26 & InN\\_SP8\\_PAX6         & 2 & 0.00610 & 1.4349994 &  3.283874 & 0.05740000 & 0.4753021 & FALSE & ASD HC65\\\\\n",
       "\t27 & InN\\_SP8\\_CRH          & 2 & 0.00615 & 1.5619382 &  3.697464 & 0.05740000 & 0.6605383 & FALSE & ASD HC65\\\\\n",
       "\t28 & ExN\\_SOX5\\_SYT6        & 2 & 0.00695 & 1.4583315 &  3.115525 & 0.05987692 & 0.5573811 & FALSE & ASD HC65\\\\\n",
       "\t29 & ExN\\_CUX2\\_ACTN2       & 2 & 0.00800 & 1.4812210 &  3.141101 & 0.06160000 & 0.6831864 & FALSE & ASD HC65\\\\\n",
       "\t30 & InN\\_LHX6\\_DCN         & 2 & 0.00825 & 1.4817412 &  3.385368 & 0.06160000 & 0.5407997 & FALSE & ASD HC65\\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t4543 & InN\\_NKX2\\_1\\_CCND2       & 2 & 0.79680 & 0.8517079 & -0.8025895 & 0.9651216 & 0.1490730 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4544 & InN\\_SP8\\_MEIS2\\_RND3     & 2 & 0.80190 & 0.8777507 & -0.8198304 & 0.9651216 & 0.1538165 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4545 & InN\\_LHX6\\_CRABP1        & 2 & 0.80340 & 0.8372672 & -0.5947048 & 0.9651216 & 0.2010952 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4546 & oIPC\\_DLL1              & 2 & 0.80530 & 0.8526240 & -0.7419515 & 0.9651216 & 0.1958917 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4547 & RB\\_HBA1\\_HBE1           & 2 & 0.80855 & 0.5806570 & -0.5761092 & 0.9651216 & 0.1073040 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4548 & InN\\_SP8\\_PAX6           & 2 & 0.82090 & 0.8485480 & -0.8738893 & 0.9651216 & 0.1615550 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4549 & InN\\_NR2F2\\_VIP          & 2 & 0.82375 & 0.8023415 & -0.8025472 & 0.9651216 & 0.2467083 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4550 & Cls\\_FGF17\\_LGI1         & 2 & 0.82770 & 0.8245840 & -0.7245079 & 0.9651216 & 0.1882434 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4551 & Astro\\_GFAP             & 2 & 0.82920 & 0.7804154 & -0.8778759 & 0.9651216 & 0.2374910 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4552 & InN\\_NR2F2\\_LAMP5        & 2 & 0.83095 & 0.8157384 & -0.8706271 & 0.9651216 & 0.2165941 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4553 & IPC\\_RSPO3\\_NEUROG1      & 2 & 0.83300 & 0.7089259 & -0.5987480 & 0.9651216 & 0.1830551 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4554 & InN\\_FOXP2\\_TSHZ1        & 2 & 0.85075 & 0.8382707 & -0.9043727 & 0.9651216 & 0.1836489 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4555 & Oligo\\_MBP\\_PLP1         & 2 & 0.85480 & 0.6431800 & -0.7476012 & 0.9651216 & 0.2308018 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4556 & ExN\\_SOX5\\_KCNV1         & 2 & 0.85610 & 0.8114204 & -1.0291166 & 0.9651216 & 0.1999997 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4557 & IPC\\_EOMES\\_VIM          & 2 & 0.85810 & 0.8592431 & -1.0250441 & 0.9651216 & 0.1505864 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4558 & VLMC\\_PTGDS\\_DSP         & 2 & 0.86025 & 0.6000721 & -0.8290546 & 0.9651216 & 0.2154650 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4559 & Astro\\_MFGE8            & 2 & 0.86705 & 0.7452408 & -0.9995434 & 0.9651216 & 0.2294424 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4560 & IPC\\_EOMES\\_NHLH1\\_deep   & 2 & 0.86935 & 0.8234710 & -0.9620617 & 0.9651216 & 0.1365066 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4561 & ExN\\_SOX5\\_SYT6          & 2 & 0.87635 & 0.7978923 & -1.0443020 & 0.9651216 & 0.1755591 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4562 & ExN\\_CUX2\\_ADRA2A        & 2 & 0.87895 & 0.8161150 & -1.0427608 & 0.9651216 & 0.1961770 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4563 & RB\\_HBA1\\_SNCA           & 2 & 0.90390 & 0.7036430 & -0.6623377 & 0.9828816 & 0.1466737 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4564 & OPC\\_PDGFRA             & 2 & 0.92540 & 0.7469174 & -1.0876794 & 0.9834811 & 0.2139045 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4565 & ExN\\_TSHZ2\\_NR4A3        & 2 & 0.94015 & 0.6767839 & -1.1731889 & 0.9834811 & 0.2261227 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4566 & ExN\\_CUX2\\_PALMD         & 2 & 0.94735 & 0.7858237 & -1.3829181 & 0.9834811 & 0.1609537 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4567 & PC\\_TTR                 & 2 & 0.94895 & 0.6603859 & -0.8729626 & 0.9834811 & 0.1570861 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4568 & ExN\\_SOX5\\_NR4A2\\_GRID2   & 2 & 0.95240 & 0.7019206 & -1.2645338 & 0.9834811 & 0.1675807 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4569 & Endothelial\\_CLDN5\\_FLT1 & 2 & 0.96145 & 0.4904144 & -1.1765550 & 0.9834811 & 0.1695690 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4570 & Microglia\\_PTPRC\\_C1QC   & 2 & 0.97050 & 0.4924781 & -1.0450169 & 0.9834811 & 0.1539820 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4571 & IPC\\_EOMES\\_NHLH1\\_up     & 2 & 0.97470 & 0.7381648 & -1.4672831 & 0.9834811 & 0.1388052 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\t4572 & Ependymal\\_FOXJ1        & 2 & 0.98990 & 0.4851367 & -1.0553920 & 0.9899000 & 0.1896320 & FALSE & DISGENET::Neuroblastoma\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4572 × 9\n",
       "\n",
       "| <!--/--> | CellType &lt;chr&gt; | annotLevel &lt;int&gt; | p &lt;dbl&gt; | fold_change &lt;dbl&gt; | sd_from_mean &lt;dbl&gt; | q &lt;dbl&gt; | hit.cells &lt;dbl&gt; | geneSizeControl &lt;lgl&gt; | list &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | Excitatory_neurons   | 1 | 0.00005 | 1.6210245 |  4.949783 | 0.00075000 | 5.1582406 | FALSE | ASD HC65 |\n",
       "| 2 | Inhibitory_neurons   | 1 | 0.00025 | 1.6011505 |  4.651712 | 0.00187500 | 4.7708344 | FALSE | ASD HC65 |\n",
       "| 3 | gIPC                 | 1 | 0.05595 | 1.1455764 |  1.676427 | 0.27975000 | 3.7065873 | FALSE | ASD HC65 |\n",
       "| 4 | OPC&amp;Oligo            | 1 | 0.09410 | 1.1895992 |  1.392667 | 0.28470000 | 4.5192011 | FALSE | ASD HC65 |\n",
       "| 5 | CR                   | 1 | 0.09490 | 1.1910740 |  1.366746 | 0.28470000 | 3.8939663 | FALSE | ASD HC65 |\n",
       "| 6 | enIPC                | 1 | 0.11695 | 1.1109784 |  1.144413 | 0.29237500 | 2.7927076 | FALSE | ASD HC65 |\n",
       "| 7 | Astro                | 1 | 0.30115 | 1.0635441 |  0.460351 | 0.64532143 | 3.9141276 | FALSE | ASD HC65 |\n",
       "| 8 | inIPC                | 1 | 0.87095 | 0.8844217 | -1.075973 | 0.99990000 | 2.4041449 | FALSE | ASD HC65 |\n",
       "| 9 | RB&amp;Vas               | 1 | 0.91495 | 0.7682445 | -1.260677 | 0.99990000 | 2.9470457 | FALSE | ASD HC65 |\n",
       "| 10 | Immune               | 1 | 0.97145 | 0.6909600 | -1.608298 | 0.99990000 | 2.2556185 | FALSE | ASD HC65 |\n",
       "| 11 | Mes                  | 1 | 0.97440 | 0.7195762 | -1.577106 | 0.99990000 | 2.1893353 | FALSE | ASD HC65 |\n",
       "| 12 | PAT_related_subtypes | 1 | 0.99620 | 0.7863180 | -2.327396 | 0.99990000 | 2.2326491 | FALSE | ASD HC65 |\n",
       "| 13 | GE_NSC               | 1 | 0.99890 | 0.7601149 | -2.664843 | 0.99990000 | 2.3555541 | FALSE | ASD HC65 |\n",
       "| 14 | dorsal_NSC           | 1 | 0.99980 | 0.7300410 | -3.089397 | 0.99990000 | 2.0313232 | FALSE | ASD HC65 |\n",
       "| 15 | Patterning_centers   | 1 | 0.99990 | 0.6629693 | -3.137240 | 0.99990000 | 1.8286641 | FALSE | ASD HC65 |\n",
       "| 16 | InN_NR2F2_SP8        | 2 | 0.00085 | 1.4028790 |  3.810475 | 0.04666667 | 0.4731544 | FALSE | ASD HC65 |\n",
       "| 17 | InN_NR2F2_CRH        | 2 | 0.00130 | 1.5360277 |  3.909490 | 0.04666667 | 0.6167353 | FALSE | ASD HC65 |\n",
       "| 18 | InN_LHX6_MAF         | 2 | 0.00135 | 1.4953935 |  3.932042 | 0.04666667 | 0.4736134 | FALSE | ASD HC65 |\n",
       "| 19 | ExN_CUX2_PALMD       | 2 | 0.00170 | 1.4938997 |  4.199320 | 0.04666667 | 0.5322803 | FALSE | ASD HC65 |\n",
       "| 20 | InN_LHX6_SST_RELN    | 2 | 0.00245 | 1.5609166 |  3.993785 | 0.04666667 | 0.5792726 | FALSE | ASD HC65 |\n",
       "| 21 | InN_LHX6_CCK         | 2 | 0.00250 | 1.4522059 |  3.749336 | 0.04666667 | 0.5620510 | FALSE | ASD HC65 |\n",
       "| 22 | ExN_SOX5_OPRK1_SULF1 | 2 | 0.00300 | 1.9609908 |  4.065703 | 0.04800000 | 1.1837654 | FALSE | ASD HC65 |\n",
       "| 23 | ExN_SOX5_ID2         | 2 | 0.00350 | 1.4128909 |  3.286578 | 0.04900000 | 0.4937898 | FALSE | ASD HC65 |\n",
       "| 24 | ExN_CUX2_ADRA2A      | 2 | 0.00530 | 1.4537623 |  3.388206 | 0.05740000 | 0.6081105 | FALSE | ASD HC65 |\n",
       "| 25 | ExN_SOX5_PALMD       | 2 | 0.00565 | 1.4095423 |  3.271409 | 0.05740000 | 0.4244082 | FALSE | ASD HC65 |\n",
       "| 26 | InN_SP8_PAX6         | 2 | 0.00610 | 1.4349994 |  3.283874 | 0.05740000 | 0.4753021 | FALSE | ASD HC65 |\n",
       "| 27 | InN_SP8_CRH          | 2 | 0.00615 | 1.5619382 |  3.697464 | 0.05740000 | 0.6605383 | FALSE | ASD HC65 |\n",
       "| 28 | ExN_SOX5_SYT6        | 2 | 0.00695 | 1.4583315 |  3.115525 | 0.05987692 | 0.5573811 | FALSE | ASD HC65 |\n",
       "| 29 | ExN_CUX2_ACTN2       | 2 | 0.00800 | 1.4812210 |  3.141101 | 0.06160000 | 0.6831864 | FALSE | ASD HC65 |\n",
       "| 30 | InN_LHX6_DCN         | 2 | 0.00825 | 1.4817412 |  3.385368 | 0.06160000 | 0.5407997 | FALSE | ASD HC65 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 4543 | InN_NKX2_1_CCND2       | 2 | 0.79680 | 0.8517079 | -0.8025895 | 0.9651216 | 0.1490730 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4544 | InN_SP8_MEIS2_RND3     | 2 | 0.80190 | 0.8777507 | -0.8198304 | 0.9651216 | 0.1538165 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4545 | InN_LHX6_CRABP1        | 2 | 0.80340 | 0.8372672 | -0.5947048 | 0.9651216 | 0.2010952 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4546 | oIPC_DLL1              | 2 | 0.80530 | 0.8526240 | -0.7419515 | 0.9651216 | 0.1958917 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4547 | RB_HBA1_HBE1           | 2 | 0.80855 | 0.5806570 | -0.5761092 | 0.9651216 | 0.1073040 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4548 | InN_SP8_PAX6           | 2 | 0.82090 | 0.8485480 | -0.8738893 | 0.9651216 | 0.1615550 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4549 | InN_NR2F2_VIP          | 2 | 0.82375 | 0.8023415 | -0.8025472 | 0.9651216 | 0.2467083 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4550 | Cls_FGF17_LGI1         | 2 | 0.82770 | 0.8245840 | -0.7245079 | 0.9651216 | 0.1882434 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4551 | Astro_GFAP             | 2 | 0.82920 | 0.7804154 | -0.8778759 | 0.9651216 | 0.2374910 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4552 | InN_NR2F2_LAMP5        | 2 | 0.83095 | 0.8157384 | -0.8706271 | 0.9651216 | 0.2165941 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4553 | IPC_RSPO3_NEUROG1      | 2 | 0.83300 | 0.7089259 | -0.5987480 | 0.9651216 | 0.1830551 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4554 | InN_FOXP2_TSHZ1        | 2 | 0.85075 | 0.8382707 | -0.9043727 | 0.9651216 | 0.1836489 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4555 | Oligo_MBP_PLP1         | 2 | 0.85480 | 0.6431800 | -0.7476012 | 0.9651216 | 0.2308018 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4556 | ExN_SOX5_KCNV1         | 2 | 0.85610 | 0.8114204 | -1.0291166 | 0.9651216 | 0.1999997 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4557 | IPC_EOMES_VIM          | 2 | 0.85810 | 0.8592431 | -1.0250441 | 0.9651216 | 0.1505864 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4558 | VLMC_PTGDS_DSP         | 2 | 0.86025 | 0.6000721 | -0.8290546 | 0.9651216 | 0.2154650 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4559 | Astro_MFGE8            | 2 | 0.86705 | 0.7452408 | -0.9995434 | 0.9651216 | 0.2294424 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4560 | IPC_EOMES_NHLH1_deep   | 2 | 0.86935 | 0.8234710 | -0.9620617 | 0.9651216 | 0.1365066 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4561 | ExN_SOX5_SYT6          | 2 | 0.87635 | 0.7978923 | -1.0443020 | 0.9651216 | 0.1755591 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4562 | ExN_CUX2_ADRA2A        | 2 | 0.87895 | 0.8161150 | -1.0427608 | 0.9651216 | 0.1961770 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4563 | RB_HBA1_SNCA           | 2 | 0.90390 | 0.7036430 | -0.6623377 | 0.9828816 | 0.1466737 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4564 | OPC_PDGFRA             | 2 | 0.92540 | 0.7469174 | -1.0876794 | 0.9834811 | 0.2139045 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4565 | ExN_TSHZ2_NR4A3        | 2 | 0.94015 | 0.6767839 | -1.1731889 | 0.9834811 | 0.2261227 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4566 | ExN_CUX2_PALMD         | 2 | 0.94735 | 0.7858237 | -1.3829181 | 0.9834811 | 0.1609537 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4567 | PC_TTR                 | 2 | 0.94895 | 0.6603859 | -0.8729626 | 0.9834811 | 0.1570861 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4568 | ExN_SOX5_NR4A2_GRID2   | 2 | 0.95240 | 0.7019206 | -1.2645338 | 0.9834811 | 0.1675807 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4569 | Endothelial_CLDN5_FLT1 | 2 | 0.96145 | 0.4904144 | -1.1765550 | 0.9834811 | 0.1695690 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4570 | Microglia_PTPRC_C1QC   | 2 | 0.97050 | 0.4924781 | -1.0450169 | 0.9834811 | 0.1539820 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4571 | IPC_EOMES_NHLH1_up     | 2 | 0.97470 | 0.7381648 | -1.4672831 | 0.9834811 | 0.1388052 | FALSE | DISGENET::Neuroblastoma |\n",
       "| 4572 | Ependymal_FOXJ1        | 2 | 0.98990 | 0.4851367 | -1.0553920 | 0.9899000 | 0.1896320 | FALSE | DISGENET::Neuroblastoma |\n",
       "\n"
      ],
      "text/plain": [
       "     CellType               annotLevel p       fold_change sd_from_mean\n",
       "1    Excitatory_neurons     1          0.00005 1.6210245    4.949783   \n",
       "2    Inhibitory_neurons     1          0.00025 1.6011505    4.651712   \n",
       "3    gIPC                   1          0.05595 1.1455764    1.676427   \n",
       "4    OPC&Oligo              1          0.09410 1.1895992    1.392667   \n",
       "5    CR                     1          0.09490 1.1910740    1.366746   \n",
       "6    enIPC                  1          0.11695 1.1109784    1.144413   \n",
       "7    Astro                  1          0.30115 1.0635441    0.460351   \n",
       "8    inIPC                  1          0.87095 0.8844217   -1.075973   \n",
       "9    RB&Vas                 1          0.91495 0.7682445   -1.260677   \n",
       "10   Immune                 1          0.97145 0.6909600   -1.608298   \n",
       "11   Mes                    1          0.97440 0.7195762   -1.577106   \n",
       "12   PAT_related_subtypes   1          0.99620 0.7863180   -2.327396   \n",
       "13   GE_NSC                 1          0.99890 0.7601149   -2.664843   \n",
       "14   dorsal_NSC             1          0.99980 0.7300410   -3.089397   \n",
       "15   Patterning_centers     1          0.99990 0.6629693   -3.137240   \n",
       "16   InN_NR2F2_SP8          2          0.00085 1.4028790    3.810475   \n",
       "17   InN_NR2F2_CRH          2          0.00130 1.5360277    3.909490   \n",
       "18   InN_LHX6_MAF           2          0.00135 1.4953935    3.932042   \n",
       "19   ExN_CUX2_PALMD         2          0.00170 1.4938997    4.199320   \n",
       "20   InN_LHX6_SST_RELN      2          0.00245 1.5609166    3.993785   \n",
       "21   InN_LHX6_CCK           2          0.00250 1.4522059    3.749336   \n",
       "22   ExN_SOX5_OPRK1_SULF1   2          0.00300 1.9609908    4.065703   \n",
       "23   ExN_SOX5_ID2           2          0.00350 1.4128909    3.286578   \n",
       "24   ExN_CUX2_ADRA2A        2          0.00530 1.4537623    3.388206   \n",
       "25   ExN_SOX5_PALMD         2          0.00565 1.4095423    3.271409   \n",
       "26   InN_SP8_PAX6           2          0.00610 1.4349994    3.283874   \n",
       "27   InN_SP8_CRH            2          0.00615 1.5619382    3.697464   \n",
       "28   ExN_SOX5_SYT6          2          0.00695 1.4583315    3.115525   \n",
       "29   ExN_CUX2_ACTN2         2          0.00800 1.4812210    3.141101   \n",
       "30   InN_LHX6_DCN           2          0.00825 1.4817412    3.385368   \n",
       "⋮    ⋮                      ⋮          ⋮       ⋮           ⋮           \n",
       "4543 InN_NKX2_1_CCND2       2          0.79680 0.8517079   -0.8025895  \n",
       "4544 InN_SP8_MEIS2_RND3     2          0.80190 0.8777507   -0.8198304  \n",
       "4545 InN_LHX6_CRABP1        2          0.80340 0.8372672   -0.5947048  \n",
       "4546 oIPC_DLL1              2          0.80530 0.8526240   -0.7419515  \n",
       "4547 RB_HBA1_HBE1           2          0.80855 0.5806570   -0.5761092  \n",
       "4548 InN_SP8_PAX6           2          0.82090 0.8485480   -0.8738893  \n",
       "4549 InN_NR2F2_VIP          2          0.82375 0.8023415   -0.8025472  \n",
       "4550 Cls_FGF17_LGI1         2          0.82770 0.8245840   -0.7245079  \n",
       "4551 Astro_GFAP             2          0.82920 0.7804154   -0.8778759  \n",
       "4552 InN_NR2F2_LAMP5        2          0.83095 0.8157384   -0.8706271  \n",
       "4553 IPC_RSPO3_NEUROG1      2          0.83300 0.7089259   -0.5987480  \n",
       "4554 InN_FOXP2_TSHZ1        2          0.85075 0.8382707   -0.9043727  \n",
       "4555 Oligo_MBP_PLP1         2          0.85480 0.6431800   -0.7476012  \n",
       "4556 ExN_SOX5_KCNV1         2          0.85610 0.8114204   -1.0291166  \n",
       "4557 IPC_EOMES_VIM          2          0.85810 0.8592431   -1.0250441  \n",
       "4558 VLMC_PTGDS_DSP         2          0.86025 0.6000721   -0.8290546  \n",
       "4559 Astro_MFGE8            2          0.86705 0.7452408   -0.9995434  \n",
       "4560 IPC_EOMES_NHLH1_deep   2          0.86935 0.8234710   -0.9620617  \n",
       "4561 ExN_SOX5_SYT6          2          0.87635 0.7978923   -1.0443020  \n",
       "4562 ExN_CUX2_ADRA2A        2          0.87895 0.8161150   -1.0427608  \n",
       "4563 RB_HBA1_SNCA           2          0.90390 0.7036430   -0.6623377  \n",
       "4564 OPC_PDGFRA             2          0.92540 0.7469174   -1.0876794  \n",
       "4565 ExN_TSHZ2_NR4A3        2          0.94015 0.6767839   -1.1731889  \n",
       "4566 ExN_CUX2_PALMD         2          0.94735 0.7858237   -1.3829181  \n",
       "4567 PC_TTR                 2          0.94895 0.6603859   -0.8729626  \n",
       "4568 ExN_SOX5_NR4A2_GRID2   2          0.95240 0.7019206   -1.2645338  \n",
       "4569 Endothelial_CLDN5_FLT1 2          0.96145 0.4904144   -1.1765550  \n",
       "4570 Microglia_PTPRC_C1QC   2          0.97050 0.4924781   -1.0450169  \n",
       "4571 IPC_EOMES_NHLH1_up     2          0.97470 0.7381648   -1.4672831  \n",
       "4572 Ependymal_FOXJ1        2          0.98990 0.4851367   -1.0553920  \n",
       "     q          hit.cells geneSizeControl list                   \n",
       "1    0.00075000 5.1582406 FALSE           ASD HC65               \n",
       "2    0.00187500 4.7708344 FALSE           ASD HC65               \n",
       "3    0.27975000 3.7065873 FALSE           ASD HC65               \n",
       "4    0.28470000 4.5192011 FALSE           ASD HC65               \n",
       "5    0.28470000 3.8939663 FALSE           ASD HC65               \n",
       "6    0.29237500 2.7927076 FALSE           ASD HC65               \n",
       "7    0.64532143 3.9141276 FALSE           ASD HC65               \n",
       "8    0.99990000 2.4041449 FALSE           ASD HC65               \n",
       "9    0.99990000 2.9470457 FALSE           ASD HC65               \n",
       "10   0.99990000 2.2556185 FALSE           ASD HC65               \n",
       "11   0.99990000 2.1893353 FALSE           ASD HC65               \n",
       "12   0.99990000 2.2326491 FALSE           ASD HC65               \n",
       "13   0.99990000 2.3555541 FALSE           ASD HC65               \n",
       "14   0.99990000 2.0313232 FALSE           ASD HC65               \n",
       "15   0.99990000 1.8286641 FALSE           ASD HC65               \n",
       "16   0.04666667 0.4731544 FALSE           ASD HC65               \n",
       "17   0.04666667 0.6167353 FALSE           ASD HC65               \n",
       "18   0.04666667 0.4736134 FALSE           ASD HC65               \n",
       "19   0.04666667 0.5322803 FALSE           ASD HC65               \n",
       "20   0.04666667 0.5792726 FALSE           ASD HC65               \n",
       "21   0.04666667 0.5620510 FALSE           ASD HC65               \n",
       "22   0.04800000 1.1837654 FALSE           ASD HC65               \n",
       "23   0.04900000 0.4937898 FALSE           ASD HC65               \n",
       "24   0.05740000 0.6081105 FALSE           ASD HC65               \n",
       "25   0.05740000 0.4244082 FALSE           ASD HC65               \n",
       "26   0.05740000 0.4753021 FALSE           ASD HC65               \n",
       "27   0.05740000 0.6605383 FALSE           ASD HC65               \n",
       "28   0.05987692 0.5573811 FALSE           ASD HC65               \n",
       "29   0.06160000 0.6831864 FALSE           ASD HC65               \n",
       "30   0.06160000 0.5407997 FALSE           ASD HC65               \n",
       "⋮    ⋮          ⋮         ⋮               ⋮                      \n",
       "4543 0.9651216  0.1490730 FALSE           DISGENET::Neuroblastoma\n",
       "4544 0.9651216  0.1538165 FALSE           DISGENET::Neuroblastoma\n",
       "4545 0.9651216  0.2010952 FALSE           DISGENET::Neuroblastoma\n",
       "4546 0.9651216  0.1958917 FALSE           DISGENET::Neuroblastoma\n",
       "4547 0.9651216  0.1073040 FALSE           DISGENET::Neuroblastoma\n",
       "4548 0.9651216  0.1615550 FALSE           DISGENET::Neuroblastoma\n",
       "4549 0.9651216  0.2467083 FALSE           DISGENET::Neuroblastoma\n",
       "4550 0.9651216  0.1882434 FALSE           DISGENET::Neuroblastoma\n",
       "4551 0.9651216  0.2374910 FALSE           DISGENET::Neuroblastoma\n",
       "4552 0.9651216  0.2165941 FALSE           DISGENET::Neuroblastoma\n",
       "4553 0.9651216  0.1830551 FALSE           DISGENET::Neuroblastoma\n",
       "4554 0.9651216  0.1836489 FALSE           DISGENET::Neuroblastoma\n",
       "4555 0.9651216  0.2308018 FALSE           DISGENET::Neuroblastoma\n",
       "4556 0.9651216  0.1999997 FALSE           DISGENET::Neuroblastoma\n",
       "4557 0.9651216  0.1505864 FALSE           DISGENET::Neuroblastoma\n",
       "4558 0.9651216  0.2154650 FALSE           DISGENET::Neuroblastoma\n",
       "4559 0.9651216  0.2294424 FALSE           DISGENET::Neuroblastoma\n",
       "4560 0.9651216  0.1365066 FALSE           DISGENET::Neuroblastoma\n",
       "4561 0.9651216  0.1755591 FALSE           DISGENET::Neuroblastoma\n",
       "4562 0.9651216  0.1961770 FALSE           DISGENET::Neuroblastoma\n",
       "4563 0.9828816  0.1466737 FALSE           DISGENET::Neuroblastoma\n",
       "4564 0.9834811  0.2139045 FALSE           DISGENET::Neuroblastoma\n",
       "4565 0.9834811  0.2261227 FALSE           DISGENET::Neuroblastoma\n",
       "4566 0.9834811  0.1609537 FALSE           DISGENET::Neuroblastoma\n",
       "4567 0.9834811  0.1570861 FALSE           DISGENET::Neuroblastoma\n",
       "4568 0.9834811  0.1675807 FALSE           DISGENET::Neuroblastoma\n",
       "4569 0.9834811  0.1695690 FALSE           DISGENET::Neuroblastoma\n",
       "4570 0.9834811  0.1539820 FALSE           DISGENET::Neuroblastoma\n",
       "4571 0.9834811  0.1388052 FALSE           DISGENET::Neuroblastoma\n",
       "4572 0.9899000  0.1896320 FALSE           DISGENET::Neuroblastoma"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sapply(disease_lists, length)\n",
    "\n",
    "if (is.character(ctd)){load(ctd)}\n",
    "\n",
    "# Start buffer of results\n",
    "dis_results <- list()\n",
    "\n",
    "max_tries <- 10\n",
    "geneSizeControl <- F\n",
    "\n",
    "# Iterate lists of diseases\n",
    "for (dis in names(disease_lists)){\n",
    "    \n",
    "    # get genset\n",
    "    gset <- disease_lists[[dis]]\n",
    "\n",
    "    # Decide which level of annotation to analyse, we do both\n",
    "    for (level in c(1:2)){\n",
    "\n",
    "        # Start summary\n",
    "print(paste(which(names(disease_lists)==dis), length(disease_lists), sep='/'))\n",
    "        print(paste('### Disease:', dis))\n",
    "        print(paste('### GC&Length control:', geneSizeControl))\n",
    "        print(paste('### Level:', level))\n",
    "        \n",
    "        # define a name of the run and filename\n",
    "        tag <- paste(dis, level, geneSizeControl) \n",
    "        dis_results.fname <- paste0(dis.res.folder, 'EWCE_results.', reps, '_reps.', tag, '.csv')\n",
    "\n",
    "        # If results were computed, load them\n",
    "        if (file.exists(dis_results.fname)){\n",
    "            print('Loading results')\n",
    "            dis_results[[tag]] <- read.csv(dis_results.fname, row.names=1)\n",
    "        # If not, try to compute it upto max.tries\n",
    "        } else {\n",
    "            tries <- 0\n",
    "            while (!tag %in% names(dis_results) & tries < max_tries){\n",
    "                tryCatch(\n",
    "                    {\n",
    "                        tries <- tries +1\n",
    "                        res <- bootstrap_enrichment_test(\n",
    "\n",
    "                            sct_data = ctd,\n",
    "\n",
    "                            hits = gset,\n",
    "                            bg = NULL,\n",
    "\n",
    "                            genelistSpecies = 'human',\n",
    "                            sctSpecies = 'human',\n",
    "                            output_species = \"human\",\n",
    "\n",
    "                            reps = reps,\n",
    "\n",
    "                            annotLevel = level,\n",
    "                            geneSizeControl = geneSizeControl,\n",
    "                            controlledCT = NULL,\n",
    "                            mtc_method = \"BH\",\n",
    "\n",
    "                            no_cores = n.cores,\n",
    "                            sort_results = TRUE,\n",
    "                            verbose = verbose)\n",
    "\n",
    "\n",
    "                        res$results$hit.cells <- res$hit.cells[rownames(res$results)]\n",
    "                        res$results$geneSizeControl <- geneSizeControl\n",
    "                        \n",
    "                        # Save, export and delete\n",
    "                        dis_results[[tag]] <- data.frame(res$results, list=dis)\n",
    "                        write.csv(dis_results[[tag]], dis_results.fname)\n",
    "                        rm(res)\n",
    "                    },\n",
    "                    error=function(cond) {\n",
    "                        print(as.character(cond))\n",
    "\n",
    "                        print(tag)\n",
    "                        if (tries > max_tries){\n",
    "                            print('COULD NOT COMPUTE')\n",
    "                        } else {\n",
    "                            print('Repeating')\n",
    "                        }\n",
    "                        # Choose a return value in case of error\n",
    "                        return()\n",
    "                    })\n",
    "                gc()\n",
    "            }\n",
    "        }\n",
    "        print('next...')\n",
    "    }\n",
    "}\n",
    "\n",
    "# Combine results\n",
    "results <- do.call('rbind', dis_results)\n",
    "rownames(results) <- 1:nrow(results)\n",
    "results\n",
    "\n",
    "# Export results\n",
    "write.csv(results, file=EWCE_results.fname)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "b477933d-a16a-4b0d-90c7-cfeb320a9ef2",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"2023-10-13 15:46:35 CEST\"\n"
     ]
    }
   ],
   "source": [
    "print(Sys.time())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b23849d4-3f43-4252-9323-3061035cdfbd",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 4.3.1 (2023-06-16)\n",
       "Platform: x86_64-conda-linux-gnu (64-bit)\n",
       "Running under: Ubuntu 20.04.5 LTS\n",
       "\n",
       "Matrix products: default\n",
       "BLAS/LAPACK: /users/genomics/xoel/micromamba/envs/Portfolio/lib/libopenblasp-r0.3.23.so;  LAPACK version 3.11.0\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n",
       " [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n",
       " [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n",
       "[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n",
       "\n",
       "time zone: Europe/Madrid\n",
       "tzcode source: system (glibc)\n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       "[1] Seurat_4.3.0.1     SeuratObject_4.1.3 sp_2.0-0           EWCE_1.8.2        \n",
       "[5] RNOmni_1.0.1.2    \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       "  [1] RcppAnnoy_0.0.21              splines_4.3.1                \n",
       "  [3] later_1.3.1                   pbdZMQ_0.3-9                 \n",
       "  [5] bitops_1.0-7                  ggplotify_0.1.2              \n",
       "  [7] filelock_1.0.2                tibble_3.2.1                 \n",
       "  [9] polyclip_1.10-4               lifecycle_1.0.3              \n",
       " [11] rstatix_0.7.2                 globals_0.16.2               \n",
       " [13] lattice_0.21-8                MASS_7.3-60                  \n",
       " [15] backports_1.4.1               magrittr_2.0.3               \n",
       " [17] limma_3.56.2                  plotly_4.10.2                \n",
       " [19] yaml_2.3.7                    httpuv_1.6.11                \n",
       " [21] sctransform_0.4.0             spatstat.sparse_3.0-2        \n",
       " [23] reticulate_1.32.0             HGNChelper_0.8.1             \n",
       " [25] pbapply_1.7-2                 cowplot_1.1.1                \n",
       " [27] DBI_1.1.3                     RColorBrewer_1.1-3           \n",
       " [29] abind_1.4-5                   zlibbioc_1.46.0              \n",
       " [31] Rtsne_0.16                    GenomicRanges_1.52.0         \n",
       " [33] purrr_1.0.1                   BiocGenerics_0.46.0          \n",
       " [35] RCurl_1.98-1.12               yulab.utils_0.1.0            \n",
       " [37] rappdirs_0.3.3                GenomeInfoDbData_1.2.10      \n",
       " [39] IRanges_2.34.1                S4Vectors_0.38.1             \n",
       " [41] ggrepel_0.9.3                 irlba_2.3.5.1                \n",
       " [43] spatstat.utils_3.0-3          listenv_0.9.0                \n",
       " [45] tidytree_0.4.5                goftest_1.2-3                \n",
       " [47] spatstat.random_3.1-6         fitdistrplus_1.1-11          \n",
       " [49] parallelly_1.36.0             leiden_0.4.3                 \n",
       " [51] codetools_0.2-19              DelayedArray_0.26.6          \n",
       " [53] tidyselect_1.2.0              aplot_0.2.1                  \n",
       " [55] spatstat.explore_3.2-3        matrixStats_1.0.0            \n",
       " [57] stats4_4.3.1                  BiocFileCache_2.8.0          \n",
       " [59] base64enc_0.1-3               jsonlite_1.8.7               \n",
       " [61] ellipsis_0.3.2                progressr_0.14.0             \n",
       " [63] ggridges_0.5.4                survival_3.5-7               \n",
       " [65] tools_4.3.1                   treeio_1.24.3                \n",
       " [67] ica_1.0-3                     Rcpp_1.0.11                  \n",
       " [69] glue_1.6.2                    gridExtra_2.3                \n",
       " [71] MatrixGenerics_1.12.2         GenomeInfoDb_1.36.1          \n",
       " [73] IRdisplay_1.1                 dplyr_1.1.2                  \n",
       " [75] BiocManager_1.30.21.1         fastmap_1.1.1                \n",
       " [77] fansi_1.0.4                   digest_0.6.33                \n",
       " [79] R6_2.5.1                      mime_0.12                    \n",
       " [81] gridGraphics_0.5-1            colorspace_2.1-0             \n",
       " [83] scattermore_1.2               tensor_1.5                   \n",
       " [85] spatstat.data_3.0-1           RSQLite_2.3.1                \n",
       " [87] utf8_1.2.3                    tidyr_1.3.0                  \n",
       " [89] generics_0.1.3                data.table_1.14.8            \n",
       " [91] httr_1.4.6                    htmlwidgets_1.6.2            \n",
       " [93] S4Arrays_1.0.5                uwot_0.1.16                  \n",
       " [95] pkgconfig_2.0.3               gtable_0.3.4                 \n",
       " [97] blob_1.2.4                    lmtest_0.9-40                \n",
       " [99] SingleCellExperiment_1.22.0   XVector_0.40.0               \n",
       "[101] htmltools_0.5.5               carData_3.0-5                \n",
       "[103] scales_1.2.1                  Biobase_2.60.0               \n",
       "[105] png_0.1-8                     ggfun_0.1.3                  \n",
       "[107] reshape2_1.4.4                uuid_1.1-0                   \n",
       "[109] nlme_3.1-163                  curl_5.0.1                   \n",
       "[111] zoo_1.8-12                    repr_1.1.6                   \n",
       "[113] cachem_1.0.8                  stringr_1.5.0                \n",
       "[115] BiocVersion_3.17.1            KernSmooth_2.23-22           \n",
       "[117] miniUI_0.1.1.1                parallel_4.3.1               \n",
       "[119] AnnotationDbi_1.62.2          pillar_1.9.0                 \n",
       "[121] grid_4.3.1                    vctrs_0.6.3                  \n",
       "[123] RANN_2.6.1                    promises_1.2.0.1             \n",
       "[125] ggpubr_0.6.0                  car_3.1-2                    \n",
       "[127] dbplyr_2.3.3                  xtable_1.8-4                 \n",
       "[129] cluster_2.1.4                 evaluate_0.21                \n",
       "[131] orthogene_1.6.1               cli_3.6.1                    \n",
       "[133] compiler_4.3.1                rlang_1.1.1                  \n",
       "[135] crayon_1.5.2                  grr_0.9.5                    \n",
       "[137] future.apply_1.11.0           ggsignif_0.6.4               \n",
       "[139] gprofiler2_0.2.2              plyr_1.8.8                   \n",
       "[141] fs_1.6.3                      stringi_1.7.12               \n",
       "[143] deldir_1.0-9                  viridisLite_0.4.2            \n",
       "[145] ewceData_1.8.0                babelgene_22.9               \n",
       "[147] munsell_0.5.0                 Biostrings_2.68.1            \n",
       "[149] lazyeval_0.2.2                spatstat.geom_3.2-5          \n",
       "[151] homologene_1.4.68.19.3.27     Matrix_1.6-1                 \n",
       "[153] IRkernel_1.3.2                ExperimentHub_2.8.1          \n",
       "[155] patchwork_1.1.3               bit64_4.0.5                  \n",
       "[157] future_1.33.0                 ggplot2_3.4.3                \n",
       "[159] KEGGREST_1.40.0               shiny_1.7.4.1                \n",
       "[161] SummarizedExperiment_1.30.2   interactiveDisplayBase_1.38.0\n",
       "[163] AnnotationHub_3.8.0           ROCR_1.0-11                  \n",
       "[165] igraph_1.5.0.1                broom_1.0.5                  \n",
       "[167] memoise_2.0.1                 ggtree_3.8.2                 \n",
       "[169] bit_4.0.5                     ape_5.7-1                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df7d9593-822d-461e-9427-6e0e8a70bea2",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Portfolio (R)",
   "language": "R",
   "name": "portfolio-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
