{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "36071ef1-cbe2-48b6-abf0-3e00b1b57293",
   "metadata": {},
   "source": [
    "### setup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e786234b-6021-41e4-bb95-9a92d68a1231",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "library(dplyr)\n",
    "\n",
    "library(ggplot2)\n",
    "library(viridis)\n",
    "library(ggpubr)\n",
    "\n",
    "fig <- function(width, heigth){\n",
    " options(repr.plot.width = width, repr.plot.height = heigth)\n",
    " }\n",
    "options(dplyr.summarise.inform = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc459b96-6641-43de-9b4f-92c222716994",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "wd <- '~/codebases/MacBrainDev/'\n",
    "data.dir <- 'data/'\n",
    "setwd(wd)\n",
    "\n",
    "n.cores <- as.numeric(Sys.getenv('SLURM_CPUS_PER_TASK'))\n",
    "n.cores <- if (is.na(n.cores)) parallel::detectCores() else n.cores\n",
    "\n",
    "\n",
    "\n",
    "# Define filename\n",
    "base.name <- 'All.MNN.v1.org.fct'\n",
    "indata.fname <- paste0(data.dir, base.name, '.rds')\n",
    "\n",
    "gene.sets <- readRDS(paste0(data.dir, 'ewce_important_genesets.rds'))\n",
    "\n",
    "all.genes <- gene.sets$hs.data.genes\n",
    "\n",
    "dis.genes.list <- gene.sets$disease_lists.data\n",
    "\n",
    "grupos <- readRDS(paste0(data.dir, 'ewce_exp_groups.rds'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f8abaec6-1e2c-4f17-acf4-4b0201cd6fb3",
   "metadata": {},
   "outputs": [],
   "source": [
    "Diseases <- readRDS(paste0(data.dir, 'DiseaseListsInfo.rds'))\n",
    "\n",
    "disease.listnames <- names(Diseases$alphabetical)\n",
    "\n",
    "disease.info <- as.data.frame(do.call('rbind', lapply(strsplit(disease.listnames, split='::', fixed=T), function(x){setNames(if (length(x)==1) c('GWAS', x) else x, c('Disease.group', 'Disease.name'))})))\n",
    "disease.info$Disease.listname <- disease.listnames                                                                                                                             \n",
    "disease.info$Disease.group <- factor(disease.info$Disease.group, levels=c('GWAS', 'MAGMA', 'DISGENET'))\n",
    "disease.info$Disease.name <- factor(disease.info$Disease.name, levels=sort(unique(disease.info$Disease.name)))\n",
    "\n",
    "disease.names <- setNames(disease.info$Disease.name, disease.info$Disease.listname)\n",
    "disease.groups <- setNames(disease.info$Disease.group, disease.info$Disease.listname)\n",
    "disease.groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1248f0a8-6391-42bd-b71b-b6250a566953",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# Load data\n",
    "indata <- readRDS(indata.fname)\n",
    "# Print summary\n",
    "indata\n",
    "\n",
    "clusters <- data.frame(unique(indata@meta.data[,c('subclass','subtype')]))\n",
    "rownames(clusters) <- clusters$subtype\n",
    "clusters\n",
    "\n",
    "rm('indata')\n",
    "gc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d8e0b4c-1e6f-48aa-8ed5-6c2761c71ae4",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# files <- grep('expression_data', dir(path = data.dir), value=T, fixed=T)\n",
    "# files <- grep('test|Report|rds$', files, value = T, invert = T)\n",
    "# files <- grep('ds100', files, value = T, invert = T, fixed=T)\n",
    "# files <- paste0(data.dir, files)\n",
    "# files <- split(files, \n",
    "#                sapply(strsplit(files, split='.', fixed=T), function(x) {rev(x)[2]}))\n",
    "#                       \n",
    "# files <- lapply(files,function(x){\n",
    "#     as.list(setNames(x, sapply(strsplit(x, split='.', fixed=T), function(y){y[2]})))\n",
    "# })\n",
    "# files <- files[sapply(files, function(x){'DotPlot' %in% names(x)})]\n",
    "# files <- files['PCs_1']\n",
    "# files\n",
    "\n",
    "files <- list(\n",
    "    PCs_1 = list(\n",
    "        DotPlot = 'data/Disease_expression_data.DotPlot.PCs_1.csv',\n",
    "        Markers ='data/Disease_expression_data.Markers.PCs_1.csv'))\n",
    "\n",
    "files"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "406accc6-c798-4a74-8160-17b19cd968d0",
   "metadata": {
    "tags": []
   },
   "source": [
    "### ranges of expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e4178dc-f3f3-4f2f-9496-b65c56889cab",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "get.dp.data <- function(fname, \n",
    "                        expressed.over.pct=5, \n",
    "                        dis...cluster...marker...filter=NULL,\n",
    "                        annot.disease=T){\n",
    "    \n",
    "    # Read data\n",
    "    dp.data <- read.csv(fname, row.names=1)\n",
    "    # Change \"id\" -> \"cluster\"\n",
    "    colnames(dp.data) <- gsub('id', 'cluster', colnames(dp.data), fixed=T)\n",
    "    \n",
    "    # Filter data using disease, cluster and marker gene\n",
    "    ## Filters are used as a pasted string of the three variables using \"...\" as separator\n",
    "    ## Used for gather expression of specific subsets in odds.ratio calculations\n",
    "    if (!is.null(dis...cluster...marker...filter)){\n",
    "        dis.cluster.gene <- apply(dp.data[, c('Disease.listname', 'cluster', 'features.plot')], 1, paste, collapse='...')\n",
    "        dp.data <- dp.data[dis.cluster.gene %in% dis...cluster...marker...filter, ]\n",
    "    }\n",
    " \n",
    "    # Summarise marker info in a df\n",
    "    df <- dp.data %>% \n",
    "        group_by(cluster, Disease.listname) %>%\n",
    "        summarize(\n",
    "            median.pct.exp=median(pct.exp), # % cells: median\n",
    "            median.avg.exp=median(avg.exp), # avg expression: median\n",
    "            median.avg.exp.scaled=median(avg.exp.scaled, na.rm=T), # avg scaled expression: median\n",
    "            mean.pct.exp=mean(pct.exp), # % cells: mean\n",
    "            mean.avg.exp=mean(avg.exp), # avg expression: mean\n",
    "            mean.avg.exp.scaled=mean(avg.exp.scaled, na.rm=T), # avg scaled expression: mean\n",
    "            n.pct.exp.over=sum(pct.exp>expressed.over.pct), # Number of genes over threshold\n",
    "            fr.pct.exp.over=mean(pct.exp>expressed.over.pct), # Fraction of genes over threshold\n",
    "            n.genes=length(unique(features.plot))) # Total number of genes per cluster and dislist\n",
    "\n",
    "    # Add info about disease lists\n",
    "    if (annot.disease){\n",
    "        df$Disease.name <- disease.names[df$Disease.listname]\n",
    "        df$Disease.group <- disease.groups[df$Disease.listname]\n",
    "        df <- df[!is.na(df$Disease.name),]\n",
    "    }\n",
    "    \n",
    "    return(df)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f6ad77e-e09c-4765-829f-87ef8b9b8872",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "# check ranges of expression\n",
    "\n",
    "df <- get.dp.data(files$PCs_1$DotPlot, 10, dis...cluster...marker...filter = NULL)\n",
    "\n",
    "df$cluster <- factor(df$cluster, grupos$PCs_1$idents)\n",
    "dff <- as.data.frame(df %>% group_by(cluster) %>% summarise(mean=mean(fr.pct.exp.over), sd=sd(fr.pct.exp.over), min=min(fr.pct.exp.over), max=max(fr.pct.exp.over)))\n",
    "rownames(dff) <- as.character(dff$cluster)\n",
    "dff$cluster <- NULL\n",
    "dff <- dff * 100\n",
    "dff$range <- paste0(round(dff$mean,2), '% (+/-',round(dff$sd,2)/2, '%)')\n",
    "dff\n",
    "# write.csv(dff, 'DiseaseGenesExpression/PCs_1/Disease-risk gene expression plot.min.pct.10.PctExpressionRanges.csv', row.names=T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a5c9c544-da6c-47b6-835b-9f9972d6ef5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "range(dff$mean)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f10c2eaf-3bf3-4680-bbaa-fa558eee4379",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Fraction of diseases genes expressed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "36007bca-a6e5-4674-98f7-a73103cc4f0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "p1.opt.a <- function(df, facet.type=F){\n",
    "    \n",
    "    p <- ggplot(df, aes(x=cluster, \n",
    "                        y=factor(Disease.name, sort(unique(Disease.name), decreasing = T)),\n",
    "                        label=ifelse(n.pct.exp.over!=0, n.pct.exp.over, NA),\n",
    "                        color=median.avg.exp,\n",
    "                        size=ifelse(n.pct.exp.over==0, NA, fr.pct.exp.over))) +\n",
    "    \n",
    "    guides(color = guide_colorbar(title.position = \"top\"),\n",
    "           size = guide_legend(title.position = \"top\")) +\n",
    "    \n",
    "    geom_point(shape=16) +\n",
    "    \n",
    "    geom_text(size=2, color='black') + \n",
    "    \n",
    "    # scale_color_viridis() +\n",
    "    scale_color_gradient(low = 'lightgrey', high='red') +\n",
    "    \n",
    "    labs(x='Cluster', \n",
    "         y='Disease', \n",
    "         size='% Disease-risk genes expressed', \n",
    "         color='Median average expression', \n",
    "         label='# Disease-risk genes expressed') +\n",
    "    \n",
    "    facet_grid(rows=vars(Disease.group), \n",
    "               cols=if (facet.type) vars(subclass) else NULL,\n",
    "               scales='free', space='free') +\n",
    "    \n",
    "    scale_size_continuous(range=c(2,7)) +\n",
    "\n",
    "    theme_pubr(base_family='Arial', legend = 'bottom') + labs_pubr() + \n",
    "    theme(text = element_text(family = 'ArialMT')) +\n",
    "\n",
    "    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +\n",
    "\n",
    "    theme(panel.background = element_rect(fill = \"#F2F2F2\"),\n",
    "          panel.grid.major = element_line(size = 0.5, linetype = 'solid', colour = \"darkgrey\"))\n",
    "\n",
    "\n",
    "    \n",
    "    \n",
    "    return(p)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e326be42-5ce8-4cae-b2eb-d27e3cb536a4",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "lapply(grupos, function(x) length(x$idents))\n",
    "\n",
    "wids <- setNames(c(11, 30, 8, 7, 8, 14, 10), names(grupos))\n",
    "wids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8bcd82e-8916-430d-a46b-9336728ab7bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "plot.label.1 <- 'DiseaseGenesExpression'\n",
    "\n",
    "min.pct <- 10\n",
    "g.plot <- 'PCs_1'\n",
    "\n",
    "\n",
    "\n",
    "message(as.character(min.pct))\n",
    "print(g.plot)\n",
    "        \n",
    "\n",
    "# Get table\n",
    "df <- get.dp.data(files[[g.plot]]$DotPlot, expressed.over.pct = min.pct)\n",
    "        \n",
    "# Set titles and labels and so on.\n",
    "subtitle <- paste0(gsub(pattern = '_', replacement = ' ', fixed = T, x = g.plot),\n",
    "                   \" (min. exp.: \", min.pct, \"% cells)\")\n",
    "wid <- wids[g.plot]\n",
    "        \n",
    "# Set type and subtype\n",
    "if (grupos[[g.plot]]$group.by == 'subclass'){\n",
    "\n",
    "    facet.type <- F\n",
    "\n",
    "    df$cluster <- droplevels(factor(df$cluster, levels=levels(clusters$subclass)))\n",
    "\n",
    "} else {\n",
    "\n",
    "    facet.type <- T\n",
    "\n",
    "    df$subclass <- droplevels(clusters[df$cluster, 'subclass'])\n",
    "\n",
    "    # Set PC order\n",
    "    if (grepl('PCs_', x=g.plot, fixed=T)){\n",
    "        df$cluster <- droplevels(factor(df$cluster, levels=grupos[[g.plot]]$idents))\n",
    "    } else {\n",
    "        df$cluster <- droplevels(factor(df$cluster, levels=rownames(clusters)))\n",
    "    }\n",
    "}\n",
    "\n",
    "table.fname <- file.path(plot.label.1, g.plot,\n",
    "                   paste('Disease-risk gene expression plot', 'min.pct', min.pct, 'csv', sep='.'))\n",
    "dir.create(dirname(table.fname), showWarnings = F, recursive=T)\n",
    "write.csv(df, table.fname)\n",
    "        \n",
    "\n",
    "\n",
    "# Get plot\n",
    "exp.plot <- p1.opt.a(df, facet.type = facet.type)        \n",
    "\n",
    "# Label plot\n",
    "exp.plot <- exp.plot + ggtitle(label = \"Disease-risk genes' expression\", subtitle=subtitle) +  xlab(paste('Cell', grupos[[g.plot]]$group.by))\n",
    "# Rotate facet labels\n",
    "if (g.plot == 'All_subclusters'){\n",
    "    exp.plot <- exp.plot + theme(strip.text.x = element_text(angle = 90, vjust = 0.5, hjust=0.5))\n",
    "}\n",
    "\n",
    "\n",
    "# Save\n",
    "fname <- file.path(plot.label.1, g.plot,\n",
    "                   paste('Disease-risk gene expression plot', 'min.pct', min.pct, 'pdf', sep='.'))\n",
    "\n",
    "ggsave(filename = fname, plot = exp.plot, width = wid, height=15, useDingbats=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "67868966-2bd0-4b75-9937-199b6d8017bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "p1 <- exp.plot"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fd1595f2-77d3-4e7e-a9b5-7e14518b1b8a",
   "metadata": {},
   "source": [
    "## Disease-risk genes overlap with markers"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ea45d99-7fb2-47cf-adc1-23899518446f",
   "metadata": {},
   "source": [
    "### Odds ratios"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe8602c8-a9e3-46bb-9244-f7eaa2340159",
   "metadata": {},
   "outputs": [],
   "source": [
    "get.prop.table <- function(fname, no.expression.pct.limit=0.10, expression.pct.limit=0.10, dpdata.fname=NULL){\n",
    "    \n",
    "    mrk.data <- read.csv(fname, row.names=1)\n",
    "\n",
    "    mrk.data$exp.1 <- ifelse(mrk.data$pct.1 < no.expression.pct.limit, 'Not expressed', ifelse(mrk.data$pct.1 >= expression.pct.limit, 'Expressed', NA))\n",
    "    mrk.data$exp.2 <- ifelse(mrk.data$pct.2 < no.expression.pct.limit, 'Not expressed', ifelse(mrk.data$pct.2 >= expression.pct.limit, 'Expressed', NA))\n",
    "\n",
    "    # print(table(mrk.data[,c('exp.1', 'exp.2')]))\n",
    "    mrk.data.filt <- filter(mrk.data, p_val_adj <=0.05 & exp.1 == 'Expressed' & exp.2 == 'Not expressed')\n",
    "\n",
    "    marker.list <- split(mrk.data.filt$gene, mrk.data.filt$cluster)\n",
    "\n",
    "    odd.table <- get.odds(marker.list)\n",
    "    \n",
    "    if(sum(odd.table$n)<1){return(NULL)}\n",
    "    \n",
    "    if (!is.null(dpdata.fname)){\n",
    "\n",
    "        dis.id.gene.filter <- tidyr::separate_rows(\n",
    "            data = odd.table,\n",
    "            risk.markers, \n",
    "            sep = ',')\n",
    "        dis.id.gene.filter <- dis.id.gene.filter[, c('Disease.listname', 'cluster', 'risk.markers')]\n",
    "        dis.id.gene.filter <- dis.id.gene.filter[dis.id.gene.filter$risk.markers!='',]\n",
    "        dis.id.gene.filter <- apply(dis.id.gene.filter, 1, paste, collapse='...')\n",
    "        \n",
    "        # dpdata <- get.dp.data(dpdata.fname, 10, dis...cluster...marker...filter = dis.id.gene.filter, annot.disease=F)\n",
    "        dpdata <- get.dp.data(dpdata.fname, expression.pct.limit*100, dis...cluster...marker...filter = dis.id.gene.filter, annot.disease=F)\n",
    "        odd.table <- merge(odd.table,dpdata,all=T, by=c('cluster', 'Disease.listname'))\n",
    "    }\n",
    "    \n",
    "    return(odd.table)\n",
    "    \n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "739e7c3d-dfd0-41c9-8e35-25bb0d7f272d",
   "metadata": {},
   "outputs": [],
   "source": [
    "get.odds <- function(marker.list, dis.genes.list=gene.sets$disease_lists.data[names(Diseases$alphabetical)]){\n",
    "    \n",
    "    pseudo.props <- c()\n",
    "    # all.tables <- list()\n",
    "\n",
    "    for (dis in names(dis.genes.list)){\n",
    "\n",
    "        dis.state <- factor(ifelse(all.genes %in% dis.genes.list[[dis]], 'Risk', 'Not risk'),\n",
    "                            levels=c('Risk', 'Not risk'))\n",
    "\n",
    "        for (st in names(marker.list)){\n",
    "\n",
    "            subt.state <- factor(ifelse(all.genes %in% marker.list[[st]], 'Marker', 'Not marker'), \n",
    "                                 levels=c('Marker', 'Not marker'))\n",
    "\n",
    "\n",
    "            d <- data.frame(dis.state, subt.state, row.names=all.genes)\n",
    "            colnames(d) <- c(dis, st)\n",
    "            risk.markers <- paste(rownames(d)[(d[,1]=='Risk') & (d[,2]=='Marker')], collapse=',')\n",
    "            contable <- table(d)\n",
    "            \n",
    "            ## Hacer fisher y sacar odd ratio\n",
    "            ### Si sale en blanco: alpha ~ p.value\n",
    "            ### Añadir el número\n",
    "            fish.test <- fisher.test(x=contable, alternative='greater')\n",
    "            # conplot <- plot(contable)\n",
    "            conmarg <- addmargins(contable)\n",
    "\n",
    "            ad.bc <- (conmarg['Risk', 'Marker'] * conmarg['Not risk', 'Not marker']) / (conmarg['Risk', 'Not marker'] * conmarg['Not risk', 'Marker'])\n",
    "            \n",
    "            pseudo.prop <- conmarg['Risk', 'Marker']/conmarg['Risk', 'Sum']/conmarg['Sum', 'Marker']\n",
    "            pseudo.props <- rbind(pseudo.props,\n",
    "                                  c(Disease.listname=dis, cluster=st,\n",
    "                                    n=conmarg['Risk', 'Marker'],\n",
    "                                    p.value=fish.test$p.value, fish.test$estimate,\n",
    "                                    ci.low=fish.test$conf.int[1], ci.high=fish.test$conf.int[2],\n",
    "                                    ad.bc = ad.bc, risk.markers=risk.markers,\n",
    "                                    log10=log10(fish.test$estimate), log2=log2(fish.test$estimate)))\n",
    "\n",
    "            # all.tables[[paste(dis, st, sep=';;')]] <- list(`Gene info`=d,\n",
    "            #                                                `Contingency table`=contable,\n",
    "            #                                                `Contingency marginals`=conmarg,\n",
    "            #                                                `Freq Risk+Marker`=pseudo.prop)\n",
    "\n",
    "        }\n",
    "    }\n",
    "\n",
    "    pseudo.props <- as.data.frame(pseudo.props)\n",
    "    pseudo.props$n <- as.numeric(pseudo.props$n)\n",
    "    pseudo.props$`odds ratio` <- as.numeric(pseudo.props$`odds ratio`)\n",
    "    pseudo.props$size.plot <- ifelse(pseudo.props$n==0, NA, pseudo.props$`odds ratio`)\n",
    "    \n",
    "    pseudo.props$Disease.name <- disease.names[pseudo.props$Disease.listname]\n",
    "    pseudo.props$Disease.group <- disease.groups[pseudo.props$Disease.listname]\n",
    "    pseudo.props <- pseudo.props[!is.na(pseudo.props$Disease.name),]\n",
    "    \n",
    "    pseudo.props$p.value <- as.numeric(pseudo.props$p.value)\n",
    "    pseudo.props$p.value.fdr <- p.adjust(pseudo.props$p.value, method = 'fdr')\n",
    "    \n",
    "    pseudo.props$significance <- ifelse(pseudo.props$p.value.fdr < 0.05, 'Sign. adj.', ifelse(pseudo.props$p.value < 0.05, 'Sign. nom.', NA))\n",
    "    \n",
    "    return(pseudo.props)\n",
    "    \n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "27bd7df7-734e-431b-ada0-e0eaaf74fd9e",
   "metadata": {},
   "source": [
    "### Plot enrichment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1143b7b3-659b-4819-bd93-e60f0db7871d",
   "metadata": {},
   "outputs": [],
   "source": [
    "p2.opt.i <- function(df, facet.type=F){\n",
    "    \n",
    "    p <- ggplot(df, aes(x=cluster, \n",
    "                        y=factor(Disease.name, sort(unique(Disease.name), decreasing = T)),\n",
    "                        label=ifelse(n==0, NA, n),\n",
    "                        size=ifelse(n==0, NA, median.avg.exp),\n",
    "                        # color=p.value<0.05,\n",
    "                        color=log2(ifelse(n==0, NA, `odds ratio`)))) +\n",
    "    \n",
    "    guides(color = guide_colorbar(title.position = \"top\"),\n",
    "           size = guide_legend(title.position = \"top\")) +\n",
    "    \n",
    "    geom_point(shape=16) +\n",
    "    \n",
    "    geom_text(size=2, color='black') + \n",
    "    \n",
    "    scale_color_gradient2(low='blue', mid='white', high = 'red', midpoint = 0) +\n",
    "    # scale_fill_continuous_divergingx(palette = 'BrBG', mid = 0, rev=T) +\n",
    "    # scale_color_manual(values=c('TRUE'='black', 'FALSE'='#F2F2F2')) + \n",
    "    \n",
    "    labs(x='Cluster', \n",
    "         y='Disease', \n",
    "         color='Odds ratio (log2)', \n",
    "         # color='p-value < 0.05',\n",
    "         size='Median average expression', \n",
    "         label='# Marker disease-risk genes') +\n",
    "    \n",
    "    facet_grid(rows=vars(Disease.group), \n",
    "               cols=if (facet.type) vars(subclass) else NULL,\n",
    "               scales='free', space='free') +\n",
    "    \n",
    "    scale_size_continuous(range=c(2,7)) +\n",
    "\n",
    "    theme_pubr(base_family='Arial', legend = 'bottom') + labs_pubr() + \n",
    "\n",
    "    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +\n",
    "    theme(text = element_text(family = 'ArialMT')) +\n",
    "\n",
    "    theme(panel.background = element_rect(fill = \"#F2F2F2\"),\n",
    "          panel.grid.major = element_line(size = 0.5, linetype = 'solid', colour = \"darkgrey\"))\n",
    "\n",
    "\n",
    "    \n",
    "    \n",
    "    return(p)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "767b8a9d-d7a8-4594-b7fe-2388f3fd9503",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "dir.create('DiseaseGenesMarkersExpression', showWarnings = F)\n",
    "plot.label.2 <- 'DiseaseGenesMarkersExpression'\n",
    "\n",
    "expression.threshold <- 0.1\n",
    "min.pct <- 0.1\n",
    "g.plot <- 'PCs_1'\n",
    "\n",
    "print(paste('group:', g.plot))\n",
    "print(paste('no exp:', expression.threshold))\n",
    "print(paste('Exp:', min.pct))\n",
    "\n",
    "# Get table\n",
    "df <- get.prop.table(fname = files[[g.plot]]$Markers, \n",
    "                     no.expression.pct.limit = expression.threshold, \n",
    "                     expression.pct.limit = min.pct, \n",
    "                     dpdata.fname = files[[g.plot]]$DotPlot)\n",
    "\n",
    "# Set titles and labels and so on.\n",
    "subtitle <- paste0(gsub(pattern = '_', replacement = ' ', fixed = T, x = g.plot),\n",
    "                   \" (Not Exp:<\", expression.threshold*100, \"% Exp:>\", min.pct*100, \"%)\")\n",
    "wid <- wids[g.plot]\n",
    "\n",
    "# Set type and subtype\n",
    "if (grupos[[g.plot]]$group.by == 'subclass'){\n",
    "\n",
    "    facet.type <- F\n",
    "\n",
    "    df$cluster <- droplevels(factor(df$cluster, levels=levels(clusters$subclass)))\n",
    "\n",
    "} else {\n",
    "\n",
    "    facet.type <- T\n",
    "\n",
    "    df$subclass <- droplevels(clusters[df$cluster, 'subclass'])\n",
    "\n",
    "    # Set PC order\n",
    "    if (grepl('PCs_', x=g.plot, fixed=T)){\n",
    "        df$cluster <- factor(df$cluster, levels=grupos[[g.plot]]$idents)\n",
    "    } else {\n",
    "        df$cluster <- factor(df$cluster, levels=rownames(clusters)[rownames(clusters)%in%grupos[[g.plot]]$idents])\n",
    "    }\n",
    "}\n",
    "table.fname <- file.path(plot.label.2, g.plot,\n",
    "                   paste('Disease-risk marker gene expression plot', 'NotExp', expression.threshold, 'Exp', min.pct, 'csv', sep='.'))\n",
    "dir.create(dirname(table.fname), showWarnings = F, recursive=T)\n",
    "write.csv(df, table.fname)\n",
    "\n",
    "\n",
    "# Get plot\n",
    "exp.plot <- p2.opt.i(df, facet.type = facet.type)\n",
    "\n",
    "# Label plot\n",
    "exp.plot <- exp.plot + \n",
    "            ggtitle(label = \"Disease-risk marker genes' expression\", subtitle=subtitle) + \n",
    "            xlab(paste('Cell', grupos[[g.plot]]$group.by))\n",
    "\n",
    "# Rotate facet labels\n",
    "if (g.plot == 'All_subclusters'){\n",
    "    exp.plot <- exp.plot + theme(strip.text.x = element_text(angle = 90, vjust = 0.5, hjust=0.5))\n",
    "}\n",
    "\n",
    "# Save\n",
    "fname <- file.path(plot.label.2, g.plot,\n",
    "                   paste('Disease-risk marker gene expression plot', 'NotExp', expression.threshold, 'Exp', min.pct, 'pdf', sep='.'))\n",
    "\n",
    "ggsave(filename = fname, plot = exp.plot, width = wid, height=15, useDingbats=FALSE)\n",
    "                \n",
    "\n",
    "\n",
    "p2 <- exp.plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90421e3b-3f61-4053-9e11-9c50e920b9a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "fig(12,12)\n",
    "combined.selected <- cowplot::plot_grid(\n",
    "    p1+ labs(x=NULL),\n",
    "    p2 + theme(axis.text.y = element_blank()) + labs(x=NULL,y=NULL), \n",
    "    align = 'h', axis = 'tb', nrow = 1, rel_widths=c(2,1.2))\n",
    "combined.selected\n",
    "\n",
    "fname <- paste('Disease-risk and marker genes expression plot', 'NotExp', 0.1, 'Exp', 0.1, 'combined','pdf', sep='.')\n",
    "\n",
    "pdf(file = fname, width = 12, height=15, useDingbats=FALSE)\n",
    "plot(combined.selected)\n",
    "dev.off()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Portfolio (R)",
   "language": "R",
   "name": "portfolio-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  },
  "toc-showtags": true
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
